{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Installing FastQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-02-19 04:17:18--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
      "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
      "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 10249221 (9.8M) [application/zip]\n",
      "Saving to: ‘fastqc_v0.11.9.zip’\n",
      "\n",
      "fastqc_v0.11.9.zip  100%[===================>]   9.77M  4.25MB/s    in 2.3s    \n",
      "\n",
      "2022-02-19 04:17:20 (4.25 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
      "\n",
      "Archive:  fastqc_v0.11.9.zip\n",
      "  inflating: FastQC/cisd-jhdf5.jar   \n",
      "   creating: FastQC/Configuration/\n",
      "  inflating: FastQC/Configuration/adapter_list.txt  \n",
      "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
      "  inflating: FastQC/Configuration/limits.txt  \n",
      "  inflating: FastQC/fastqc           \n",
      "  inflating: FastQC/fastqc_icon.ico  \n",
      "   creating: FastQC/Help/\n",
      "   creating: FastQC/Help/1 Introduction/\n",
      "   creating: FastQC/Help/1 Introduction/.svn/\n",
      "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
      "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
      "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
      "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
      "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
      "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
      "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
      "   creating: FastQC/Help/2 Basic Operations/\n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
      "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
      "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
      "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
      "   creating: FastQC/Help/3 Analysis Modules/\n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
      "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
      "  inflating: FastQC/INSTALL.txt      \n",
      "  inflating: FastQC/jbzip2-0.9.jar   \n",
      "  inflating: FastQC/LICENSE          \n",
      "  inflating: FastQC/LICENSE.txt      \n",
      "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
      "   creating: FastQC/net/\n",
      "   creating: FastQC/net/sourceforge/\n",
      "   creating: FastQC/net/sourceforge/iharder/\n",
      "   creating: FastQC/net/sourceforge/iharder/base64/\n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
      "   creating: FastQC/org/\n",
      "   creating: FastQC/org/apache/\n",
      "   creating: FastQC/org/apache/commons/\n",
      "   creating: FastQC/org/apache/commons/math3/\n",
      "   creating: FastQC/org/apache/commons/math3/analysis/\n",
      "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
      "   creating: FastQC/org/apache/commons/math3/distribution/\n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
      "   creating: FastQC/org/apache/commons/math3/exception/\n",
      "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
      "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
      "   creating: FastQC/org/apache/commons/math3/random/\n",
      "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
      "   creating: FastQC/org/apache/commons/math3/special/\n",
      "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
      "   creating: FastQC/org/apache/commons/math3/util/\n",
      "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
      "  inflating: FastQC/README.md        \n",
      "  inflating: FastQC/README.txt       \n",
      "  inflating: FastQC/RELEASE_NOTES.txt  \n",
      "  inflating: FastQC/run_fastqc.bat   \n",
      "  inflating: FastQC/sam-1.103.jar    \n",
      "   creating: FastQC/Templates/\n",
      "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
      "  inflating: FastQC/Templates/header_template.html  \n",
      "   creating: FastQC/Templates/Icons/\n",
      " extracting: FastQC/Templates/Icons/error.png  \n",
      " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
      " extracting: FastQC/Templates/Icons/tick.png  \n",
      " extracting: FastQC/Templates/Icons/warning.png  \n",
      "   creating: FastQC/uk/\n",
      "   creating: FastQC/uk/ac/\n",
      "   creating: FastQC/uk/ac/babraham/\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
     ]
    }
   ],
   "source": [
    "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
    "!unzip fastqc_v0.11.9.zip\n",
    "!chmod a+x FastQC/fastqc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Download data from ENA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-02-19 04:18:05--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
      "           => ‘SRR5836473_1.fastq.gz’\n",
      "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.138\n",
      "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.138|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
      "==> SIZE SRR5836473_1.fastq.gz ... 6856386252\n",
      "==> PASV ... done.    ==> RETR SRR5836473_1.fastq.gz ... done.\n",
      "Length: 6856386252 (6.4G) (unauthoritative)\n",
      "\n",
      "SRR5836473_1.fastq. 100%[===================>]   6.38G  4.14MB/s    in 23m 39s \n",
      "\n",
      "2022-02-19 04:41:45 (4.61 MB/s) - ‘SRR5836473_1.fastq.gz’ saved [6856386252]\n",
      "\n",
      "--2022-02-19 04:41:45--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_2.fastq.gz\n",
      "           => ‘SRR5836473_2.fastq.gz’\n",
      "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.197.74\n",
      "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
      "==> SIZE SRR5836473_2.fastq.gz ... 8127008133\n",
      "==> PASV ... done.    ==> RETR SRR5836473_2.fastq.gz ... done.\n",
      "Length: 8127008133 (7.6G) (unauthoritative)\n",
      "\n",
      "SRR5836473_2.fastq. 100%[===================>]   7.57G  4.83MB/s    in 30m 36s \n",
      "\n",
      "2022-02-19 05:12:22 (4.22 MB/s) - ‘SRR5836473_2.fastq.gz’ saved [8127008133]\n",
      "\n",
      "--2022-02-19 05:12:22--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR382/002/SRR3824222/SRR3824222_1.fastq.gz\n",
      "           => ‘SRR3824222_1.fastq.gz’\n",
      "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.197.74\n",
      "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR382/002/SRR3824222 ... done.\n",
      "==> SIZE SRR3824222_1.fastq.gz ... 8096784759\n",
      "==> PASV ... done.    ==> RETR SRR3824222_1.fastq.gz ... done.\n",
      "Length: 8096784759 (7.5G) (unauthoritative)\n",
      "\n",
      "SRR3824222_1.fastq. 100%[===================>]   7.54G  4.83MB/s    in 28m 45s \n",
      "\n",
      "2022-02-19 05:41:08 (4.48 MB/s) - ‘SRR3824222_1.fastq.gz’ saved [8096784759]\n",
      "\n",
      "--2022-02-19 05:41:08--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR382/002/SRR3824222/SRR3824222_2.fastq.gz\n",
      "           => ‘SRR3824222_2.fastq.gz’\n",
      "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.197.74\n",
      "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR382/002/SRR3824222 ... done.\n",
      "==> SIZE SRR3824222_2.fastq.gz ... 10233274189\n",
      "==> PASV ... done.    ==> RETR SRR3824222_2.fastq.gz ... done.\n",
      "Length: 10233274189 (9.5G) (unauthoritative)\n",
      "\n",
      "SRR3824222_2.fastq. 100%[===================>]   9.53G  4.76MB/s    in 35m 25s \n",
      "\n",
      "2022-02-19 06:16:33 (4.59 MB/s) - ‘SRR3824222_2.fastq.gz’ saved [10233274189]\n",
      "\n",
      "--2022-02-19 06:16:33--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/005/SRR5836475/SRR5836475_1.fastq.gz\n",
      "           => ‘SRR5836475_1.fastq.gz’\n",
      "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.197.74\n",
      "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/005/SRR5836475 ... done.\n",
      "==> SIZE SRR5836475_1.fastq.gz ... 6139527111\n",
      "==> PASV ... done.    ==> RETR SRR5836475_1.fastq.gz ... done.\n",
      "Length: 6139527111 (5.7G) (unauthoritative)\n",
      "\n",
      "SRR5836475_1.fastq. 100%[===================>]   5.72G  4.83MB/s    in 21m 7s  \n",
      "\n",
      "2022-02-19 06:37:42 (4.62 MB/s) - ‘SRR5836475_1.fastq.gz’ saved [6139527111]\n",
      "\n",
      "--2022-02-19 06:37:42--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/005/SRR5836475/SRR5836475_2.fastq.gz\n",
      "           => ‘SRR5836475_2.fastq.gz’\n",
      "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.138\n",
      "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.138|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/005/SRR5836475 ... done.\n",
      "==> SIZE SRR5836475_2.fastq.gz ... 7492692021\n",
      "==> PASV ... done.    ==> RETR SRR5836475_2.fastq.gz ... done.\n",
      "Length: 7492692021 (7.0G) (unauthoritative)\n",
      "\n",
      "SRR5836475_2.fastq. 100%[===================>]   6.98G  4.83MB/s    in 24m 53s \n",
      "\n",
      "2022-02-19 07:02:36 (4.79 MB/s) - ‘SRR5836475_2.fastq.gz’ saved [7492692021]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
    "!wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_2.fastq.gz\n",
    "!wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR382/002/SRR3824222/SRR3824222_1.fastq.gz\n",
    "!wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR382/002/SRR3824222/SRR3824222_2.fastq.gz\n",
    "!wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/005/SRR5836475/SRR5836475_1.fastq.gz\n",
    "!wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/005/SRR5836475/SRR5836475_2.fastq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run FastQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "!gunzip ./SRR3824222_1.fastq.gz -k"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR3824222_1.fastq\n",
      "Approx 5% complete for SRR3824222_1.fastq\n",
      "Approx 10% complete for SRR3824222_1.fastq\n",
      "Approx 15% complete for SRR3824222_1.fastq\n",
      "Approx 20% complete for SRR3824222_1.fastq\n",
      "Approx 25% complete for SRR3824222_1.fastq\n",
      "Approx 30% complete for SRR3824222_1.fastq\n",
      "Approx 35% complete for SRR3824222_1.fastq\n",
      "Approx 40% complete for SRR3824222_1.fastq\n",
      "Approx 45% complete for SRR3824222_1.fastq\n",
      "Approx 50% complete for SRR3824222_1.fastq\n",
      "Approx 55% complete for SRR3824222_1.fastq\n",
      "Approx 60% complete for SRR3824222_1.fastq\n",
      "Approx 65% complete for SRR3824222_1.fastq\n",
      "Approx 70% complete for SRR3824222_1.fastq\n",
      "Approx 75% complete for SRR3824222_1.fastq\n",
      "Approx 80% complete for SRR3824222_1.fastq\n",
      "Approx 85% complete for SRR3824222_1.fastq\n",
      "Approx 90% complete for SRR3824222_1.fastq\n",
      "Approx 95% complete for SRR3824222_1.fastq\n",
      "Analysis complete for SRR3824222_1.fastq\n"
     ]
    }
   ],
   "source": [
    "!./FastQC/fastqc SRR3824222_1.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "!gunzip ./SRR5836473_1.fastq.gz -k"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR5836473_1.fastq\n",
      "Approx 5% complete for SRR5836473_1.fastq\n",
      "Approx 10% complete for SRR5836473_1.fastq\n",
      "Approx 15% complete for SRR5836473_1.fastq\n",
      "Approx 20% complete for SRR5836473_1.fastq\n",
      "Approx 25% complete for SRR5836473_1.fastq\n",
      "Approx 30% complete for SRR5836473_1.fastq\n",
      "Approx 35% complete for SRR5836473_1.fastq\n",
      "Approx 40% complete for SRR5836473_1.fastq\n",
      "Approx 45% complete for SRR5836473_1.fastq\n",
      "Approx 50% complete for SRR5836473_1.fastq\n",
      "Approx 55% complete for SRR5836473_1.fastq\n",
      "Approx 60% complete for SRR5836473_1.fastq\n",
      "Approx 65% complete for SRR5836473_1.fastq\n",
      "Approx 70% complete for SRR5836473_1.fastq\n",
      "Approx 75% complete for SRR5836473_1.fastq\n",
      "Approx 80% complete for SRR5836473_1.fastq\n",
      "Approx 85% complete for SRR5836473_1.fastq\n",
      "Approx 90% complete for SRR5836473_1.fastq\n",
      "Approx 95% complete for SRR5836473_1.fastq\n",
      "Analysis complete for SRR5836473_1.fastq\n"
     ]
    }
   ],
   "source": [
    "!./FastQC/fastqc SRR5836473_1.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "!gunzip ./SRR5836475_1.fastq.gz -k"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR5836475_1.fastq\n",
      "Approx 5% complete for SRR5836475_1.fastq\n",
      "Approx 10% complete for SRR5836475_1.fastq\n",
      "Approx 15% complete for SRR5836475_1.fastq\n",
      "Approx 20% complete for SRR5836475_1.fastq\n",
      "Approx 25% complete for SRR5836475_1.fastq\n",
      "Approx 30% complete for SRR5836475_1.fastq\n",
      "Approx 35% complete for SRR5836475_1.fastq\n",
      "Approx 40% complete for SRR5836475_1.fastq\n",
      "Approx 45% complete for SRR5836475_1.fastq\n",
      "Approx 50% complete for SRR5836475_1.fastq\n",
      "Approx 55% complete for SRR5836475_1.fastq\n",
      "Approx 60% complete for SRR5836475_1.fastq\n",
      "Approx 65% complete for SRR5836475_1.fastq\n",
      "Approx 70% complete for SRR5836475_1.fastq\n",
      "Approx 75% complete for SRR5836475_1.fastq\n",
      "Approx 80% complete for SRR5836475_1.fastq\n",
      "Approx 85% complete for SRR5836475_1.fastq\n",
      "Approx 90% complete for SRR5836475_1.fastq\n",
      "Approx 95% complete for SRR5836475_1.fastq\n",
      "Analysis complete for SRR5836475_1.fastq\n"
     ]
    }
   ],
   "source": [
    "!./FastQC/fastqc SRR5836475_1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Installing Bismark with dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/lexa/anaconda3/envs/bio\n",
      "\n",
      "  added / updated specs:\n",
      "    - bismark\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    bowtie2-2.3.5.1            |   py36he513fc3_0        12.0 MB  bioconda\n",
      "    samtools-1.7               |                1         1.0 MB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:        13.0 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  bismark            bioconda/noarch::bismark-0.23.1-hdfd78af_0\n",
      "  bowtie2            bioconda/linux-64::bowtie2-2.3.5.1-py36he513fc3_0\n",
      "  bzip2              pkgs/main/linux-64::bzip2-1.0.8-h7b6447c_0\n",
      "  c-ares             pkgs/main/linux-64::c-ares-1.18.1-h7f8727e_0\n",
      "  curl               pkgs/main/linux-64::curl-7.80.0-h7f8727e_0\n",
      "  hisat2             bioconda/linux-64::hisat2-2.2.1-h1b792b2_3\n",
      "  krb5               pkgs/main/linux-64::krb5-1.19.2-hac12032_0\n",
      "  libcurl            pkgs/main/linux-64::libcurl-7.80.0-h0b77cf5_0\n",
      "  libedit            pkgs/main/linux-64::libedit-3.1.20210910-h7f8727e_0\n",
      "  libev              pkgs/main/linux-64::libev-4.33-h7f8727e_1\n",
      "  libgcc             pkgs/main/linux-64::libgcc-7.2.0-h69d50b8_2\n",
      "  libnghttp2         pkgs/main/linux-64::libnghttp2-1.46.0-hce63b2e_0\n",
      "  libssh2            pkgs/main/linux-64::libssh2-1.9.0-h1ba5d50_1\n",
      "  perl               pkgs/main/linux-64::perl-5.26.2-h14c3975_0\n",
      "  samtools           bioconda/linux-64::samtools-1.7-1\n",
      "  tbb                pkgs/main/linux-64::tbb-2021.5.0-hd09550d_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "bowtie2-2.3.5.1      | 12.0 MB   | ##################################### | 100% \n",
      "samtools-1.7         | 1.0 MB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n"
     ]
    }
   ],
   "source": [
    "!conda install -c bioconda bismark -y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: failed with initial frozen solve. Retrying with flexible solve.\n",
      "Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.\n",
      "Collecting package metadata (repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/lexa/anaconda3/envs/bio\n",
      "\n",
      "  added / updated specs:\n",
      "    - openssl=1.0\n",
      "\n",
      "\n",
      "The following packages will be REMOVED:\n",
      "\n",
      "  krb5-1.19.2-hac12032_0\n",
      "  libcurl-7.80.0-h0b77cf5_0\n",
      "  libnghttp2-1.46.0-hce63b2e_0\n",
      "\n",
      "The following packages will be DOWNGRADED:\n",
      "\n",
      "  cffi                                1.14.6-py36h400218f_0 --> 1.14.0-py36h2e261b9_0\n",
      "  curl                                    7.80.0-h7f8727e_0 --> 7.55.1-h78862de_4\n",
      "  glib                                    2.69.1-h4ff587b_1 --> 2.63.1-h5a9c865_0\n",
      "  gst-plugins-base                        1.14.0-h8213a91_2 --> 1.14.0-hbbd80ab_1\n",
      "  gstreamer                               1.14.0-h28cd5cc_2 --> 1.14.0-hb453b48_1\n",
      "  libffi                                     3.3-he6710b0_2 --> 3.2.1-hf484d3e_1007\n",
      "  libssh2                                  1.9.0-h1ba5d50_1 --> 1.8.0-h9cfc8f7_4\n",
      "  openssl                                 1.1.1m-h7f8727e_0 --> 1.0.2u-h7b6447c_0\n",
      "  python                                  3.6.13-h12debd9_1 --> 3.6.6-h6e4f718_2\n",
      "  qt                                       5.9.7-h5867ecd_1 --> 5.9.6-h8703b6f_2\n",
      "  readline                                 8.1.2-h7f8727e_1 --> 7.0-h7b6447c_5\n",
      "  sqlite                                  3.37.2-hc218d9a_0 --> 3.33.0-h62c20be_0\n",
      "\n",
      "\n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n"
     ]
    }
   ],
   "source": [
    "!conda install -c bioconda openssl=1.0 -y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Usage: samtools view [options] <in.bam>|<in.sam>|<in.cram> [region ...]\r\n",
      "\r\n",
      "Options:\r\n",
      "  -b       output BAM\r\n",
      "  -C       output CRAM (requires -T)\r\n",
      "  -1       use fast BAM compression (implies -b)\r\n",
      "  -u       uncompressed BAM output (implies -b)\r\n",
      "  -h       include header in SAM output\r\n",
      "  -H       print SAM header only (no alignments)\r\n",
      "  -c       print only the count of matching records\r\n",
      "  -o FILE  output file name [stdout]\r\n",
      "  -U FILE  output reads not selected by filters to FILE [null]\r\n",
      "  -t FILE  FILE listing reference names and lengths (see long help) [null]\r\n",
      "  -L FILE  only include reads overlapping this BED FILE [null]\r\n",
      "  -r STR   only include reads in read group STR [null]\r\n",
      "  -R FILE  only include reads with read group listed in FILE [null]\r\n",
      "  -q INT   only include reads with mapping quality >= INT [0]\r\n",
      "  -l STR   only include reads in library STR [null]\r\n",
      "  -m INT   only include reads with number of CIGAR operations consuming\r\n",
      "           query sequence >= INT [0]\r\n",
      "  -f INT   only include reads with all  of the FLAGs in INT present [0]\r\n",
      "  -F INT   only include reads with none of the FLAGS in INT present [0]\r\n",
      "  -G INT   only EXCLUDE reads with all  of the FLAGs in INT present [0]\r\n",
      "  -s FLOAT subsample reads (given INT.FRAC option value, 0.FRAC is the\r\n",
      "           fraction of templates/read pairs to keep; INT part sets seed)\r\n",
      "  -M       use the multi-region iterator (increases the speed, removes\r\n",
      "           duplicates and outputs the reads as they are ordered in the file)\r\n",
      "  -x STR   read tag to strip (repeatable) [null]\r\n",
      "  -B       collapse the backward CIGAR operation\r\n",
      "  -?       print long help, including note about region specification\r\n",
      "  -S       ignored (input format is auto-detected)\r\n",
      "      --input-fmt-option OPT[=VAL]\r\n",
      "               Specify a single input file format option in the form\r\n",
      "               of OPTION or OPTION=VALUE\r\n",
      "  -O, --output-fmt FORMAT[,OPT[=VAL]]...\r\n",
      "               Specify output format (SAM, BAM, CRAM)\r\n",
      "      --output-fmt-option OPT[=VAL]\r\n",
      "               Specify a single output file format option in the form\r\n",
      "               of OPTION or OPTION=VALUE\r\n",
      "  -T, --reference FILE\r\n",
      "               Reference sequence FASTA FILE [null]\r\n",
      "  -@, --threads INT\r\n",
      "               Number of additional threads to use [0]\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!samtools view"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Main work"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "import multiprocessing\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR5836473.18_18/1\t99\tchr11\t116830009\t0\t139M\t=\t116830059\t190\tTTTATTTTGTTATTGGATTTAGAGATTATTTGGCGGGAATCGGGTTTTTTTTTTTTTTTTTTATAATAGAGTGTCGTAATAGTAAAATTGAGTTTTGATTAGAGTGATTTTGATTTGGTTTCGTTTTTTTTTTCGTTCG\tFKKKKKKKKKKKKKKKK<FKKKKFKKKKKKKKKKKKKFKKKKKKFAFKKKKKKKKKKKKKKKAKFFFFKFF7FFFKFKK7AKKKKKKKKAFKKKKAFKKAFKFK,F<FAKKKAFAFFF,FKKKKKKKKKKKKFF<A77F\tNM:i:37\tMD:Z:0C1C2C0C2C2C6C0C23C0C0C0C0C1C0C0C0C0C2C0C2C4C0T8G2C12C4T1C7C0C4C4C1C4C3C2C3C2\tXM:Z:h.h..hx..h..x......x.............Z......Z...hhhhh.hhhhh..hh..h....x.......Z....x............h......x.......hh....h....h.xZ...h...h..xZ..xZ.\tXR:Z:CT\tXG:Z:CT\r\n",
      "SRR5836473.18_18/1\t147\tchr11\t116830059\t0\t140M\t=\t116830009\t-190\tTTTTTTTTTTTTATAATAGAGTGTCGTAATAGTAAAATTGAGTTTTGATTAGAGTGATTTTGATTTGGTTTCGTTTTTTTTTTCGTTCGTTTAGTTTTTTTTTTGTAGTTCGAGTTGTTATTTTAGTTGAATTGTTTATT\tKKF<A,KAFA7A7,K77KF<F<A<F<7,,7,7A7,,7AFFFA<,KKKAA<<,7KAF7KKFAA<AKFKKF7<KKKKFKKKKKFFKKFFFKKKKKKKKKFKKKKKKKKFKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKF\tNM:i:39\tMD:Z:0C0C0C0C0C2C0C2C4C0T8G2C12C4T1C7C0C4C4C1C4C3C2C3C3C5C0C1C1C2T0C2C0C7C0C2C9C0C3C3\tXM:Z:hhhhh..hh..h....x.......Z....x............h......x.......hh....h....h.xZ...h...h..xZ..xZ..h.....hh.h.h...x..hxZ......hh..h.........xz...h...\tXR:Z:GA\tXG:Z:CT\r\n",
      "SRR5836473.30_30/1\t99\tchr11\t24232407\t0\t140M\t=\t24232506\t239\tGGTAAGAAGGATATATGTTTAATTATGTTTATAGTAGTTTTATTTATAATAGTTAGAAGTTGGAAAGAATTTAGATGTTTTTCAATAGAGGAATGGATATTAAAAAATGTGGTATATTTATATAATGGAGTATTATTTAG\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFFKKKKKKKKKKKKKKKKKKKFKKKKKAFKK\tNM:i:29\tMD:Z:0T10C1C3C1C0C1C6C4C2C0C13C0C5C9C0C0C5C0C0C0C4C13C0C13C5C1C12C1C2\tXM:Z:...........h.h...h.h..h......h....x..hh.............hx.....x.........hhx.....hhhh.H..x.............hh.............h.....h.h............h.x..\tXR:Z:CT\tXG:Z:CT\r\n",
      "SRR5836473.30_30/1\t147\tchr11\t24232506\t0\t140M\t=\t24232407\t-239\tTTAAAAAATGTGGTATATTTATATAATGGAGTATTATTTAGTTATTAAAAAGAATGAATTTATGAAATTTTTAGGTAAATGGATGGATTTGGAGGGTATTATTTTGAGTGAGGTAATTTAATTATAAAAGAATTTATATG\t7AKKKKFFFAA7AKFAKKFKFAKKKKKFA7KKKFAFFAKKKKKKFFKFKKKKKKKKKAKKKKKFKKKKKKKKKKKKKFKKKKKKKKKFKKKKKKKKKKKKKKA<AKFKF<AKKKKKKFKFKKKKKKKKKKKKKKKKKKFF\tNM:i:28\tMD:Z:0C0C13C5C1C12C1C2C27C0C4C0C10C0C7C2C2C0C12C0C0C3C1C7C1C1C1C0A0\tXM:Z:hh.............h.....h.h............h.x..h...........................hh....h...........hx.......h..h..hx............hhh...h.h.......h.h.h.h.\tXR:Z:GA\tXG:Z:CT\r\n",
      "SRR5836473.39_39/1\t83\tchr11\t105051790\t0\t13M1D127M\t=\t105051789\t-142\tTTAAAAACTACATATACTAATATACTAACAAACAAATAAACCAATAAAATAAAATTAAAAACCCAAAAATAAACCCACACACCTATAATCACTTAATCTTTATCAAAAAAACTAAAACCATCCAATAAAAAAAAAACACC\tKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKFKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:31\tMD:Z:0A8G3^G0G4G0G6G3G4G2G6G0G4G4G2G5G4G15G0G6G6G0A3G0G0G1G13G1G0G6G3G1\tXM:Z:.........x...h....xh......x...x....h..h......hh....h....h..h.....x....h...............hh......h......h....hhh.h.............x.hh......h.....\tXR:Z:CT\tXG:Z:GA\r\n",
      "SRR5836473.39_39/1\t163\tchr11\t105051789\t0\t14M1D126M\t=\t105051790\t142\tATTAAAAACTACATATACTAATATACTAACAAACAAATAAACCAATAAAATAAAATTAAAAACCCAAAAATAAACCCACACACCTATAATCACTTAATCTTTATCAAAAAAACTAAAACCATCCAATAAAAAAAAAACAC\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKK<FKA<FKFFKKKKKKKKKFKFKKFK,,<FA,,7<\tNM:i:32\tMD:Z:0G0A8G3^G0G4G0G6G3G4G2G6G0G4G4G2G5G4G15G0G6G6G0A3G0G0G1G13G1G0G6G3G0\tXM:Z:h.........x...h....xh......x...x....h..h......hh....h....h..h.....x....h...............hh......h......h....hhh.h.............x.hh......h....\tXR:Z:GA\tXG:Z:GA\r\n",
      "SRR5836473.41_41/1\t99\tchr11\t40185837\t0\t140M\t=\t40185875\t178\tAATTTTTTTGGGGATTGTATTGAATTTGTAGATTGTTTTTGGTAAGAGAGTTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTTTTTTTTAGAGATT\tKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKFKKKKKKKKKFKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKF7<77FF\tNM:i:28\tMD:Z:6G0A9C7C9C6C4T2C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C3C2C5C1\tXM:Z:.................h.......x.........h......h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h...h..x.....h.\tXR:Z:CT\tXG:Z:CT\r\n",
      "SRR5836473.41_41/1\t147\tchr11\t40185875\t0\t140M\t=\t40185837\t-178\tTTGGTAAGAGAGTTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTTTTTTTTAGAGATTTGAAGTTTTTATTATATAGATTTTTTATTTTTTTAGTG\tAKKKKKAAFFFKKKKKKKKKKKKKFFKKKKKKKKKKFAFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:31\tMD:Z:4C4T2C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C3C2C5C8C4C3A4C3C1C2C6T0\tXM:Z:....h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h...h..x.....h........h....h........h...h.h..h.......\tXR:Z:GA\tXG:Z:CT\r\n",
      "SRR5836473.46_46/1\t83\tchr11\t22771774\t8\t140M\t=\t22771594\t-320\tCAAATTAAAAAAAAAAAAATAAAAATATCACTTTTTACAAATAATATAATAATATATATAAATAACCCTAAAAATTCTATCACAAAACTCCTAAACCTAATAAACAACTTCAATAAAATAACTAAATATAAAATTAACTC\tKKKKAAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:26\tMD:Z:6G0G4G2G2G1C15G2G2G4G3G9G1G13C1C2G1G9G3G7G5G1G2G2G2G0G15\tXM:Z:......hh....h..h..h.................h..x..h....h...h.........h.h....................h.........h...x.......x.....x.h..h..h..xh...............\tXR:Z:CT\tXG:Z:GA\r\n",
      "SRR5836473.46_46/1\t163\tchr11\t22771594\t8\t140M\t=\t22771774\t320\tAGAATAAAAACAATCTACAACAAACCAATAACCAACATCAAAATAAATAATAAAAAACTAAAAACAATCCCACTAAAATCAAAAACTAAACAAAACTACCCACTTTCTCCCTACCTATTCAACATTATACTTAAAATCCT\tFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKAKKKKKKKKKKKKKKKKKKKKKKKKFFKFKKKKKKFKKKKKKK<AKKKKKKFFF7<A<K\tNM:i:25\tMD:Z:1T0G6G9G7G2G11G5G0G1G1G2G2G0G2G17G0G0G4G4G0G2G28G5G2G4\tXM:Z:..h......h.........x.......x..h...........h.....hh.h.h..h..xh..h.................xhh....h....hh..x............................h.....h..h....\tXR:Z:GA\tXG:Z:GA\r\n",
      "samtools view: writing to standard output failed: Broken pipe\r\n",
      "samtools view: error closing standard output: -1\r\n"
     ]
    }
   ],
   "source": [
    "!samtools view ./cell8/SRR5836473_1_bismark_bt2_pe.bam | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 3 files and 1 in-memory blocks...\n",
      "[bam_sort_core] merging from 5 files and 1 in-memory blocks...\n",
      "[bam_sort_core] merging from 9 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "input_file = [\"./cell8/SRR5836473_1_bismark_bt2_pe.bam\", \"./ICM/SRR5836475_1_bismark_bt2_pe.bam\", \"./epiblast/SRR3824222_1_bismark_bt2_pe.bam\"]\n",
    "output_file = [\"sorted_8cell.bam\", \"sorted_icm.bam\", \"sorted_epiblast.bam\"]\n",
    "\n",
    "def sort_files(input_file, output_file):\n",
    "    subprocess.call([\"samtools\", \"sort\",  input_file, \"-o\", output_file])\n",
    "    \n",
    "p1 = multiprocessing.Process(target=sort_files, args=(input_file[0], output_file[0]))\n",
    "p2 = multiprocessing.Process(target=sort_files, args=(input_file[1], output_file[1]))\n",
    "p3 = multiprocessing.Process(target=sort_files, args=(input_file[2], output_file[2]))\n",
    "\n",
    "p1.start()\n",
    "p2.start()\n",
    "p3.start()\n",
    "\n",
    "p1.join()\n",
    "p2.join()\n",
    "p3.join()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "!samtools index sorted_8cell.bam\n",
    "!samtools index sorted_icm.bam\n",
    "!samtools index sorted_epiblast.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1090\n",
      "464\n",
      "1456\n",
      "630\n",
      "2328\n",
      "1062\n"
     ]
    }
   ],
   "source": [
    "!samtools view sorted_8cell.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
    "!samtools view sorted_8cell.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l\n",
    "!samtools view sorted_icm.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
    "!samtools view sorted_icm.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l\n",
    "!samtools view sorted_epiblast.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
    "!samtools view sorted_epiblast.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "input_file = [\"./cell8/SRR5836473_1_bismark_bt2_pe.bam\", \"./ICM/SRR5836475_1_bismark_bt2_pe.bam\", \"./epiblast/SRR3824222_1_bismark_bt2_pe.bam\"]\n",
    "output_file = [\"s_8_cell\", \"s_icm\", \"s_epiblast\"]\n",
    "\n",
    "def deduplicate(input_file, output_file):\n",
    "    subprocess.call([\"deduplicate_bismark\", \"--bam\",  \"--paired\", \"-o\", output_file, input_file])\n",
    "\n",
    "p1 = multiprocessing.Process(target=deduplicate, args=(input_file[0], output_file[0]))\n",
    "p2 = multiprocessing.Process(target=deduplicate, args=(input_file[1], output_file[1]))\n",
    "p3 = multiprocessing.Process(target=deduplicate, args=(input_file[2], output_file[2]))\n",
    "\n",
    "p1.start()\n",
    "p2.start()\n",
    "p3.start()\n",
    "\n",
    "p1.join()\n",
    "p2.join()\n",
    "p3.join()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "input_file = [\"s_8_cell.deduplicated.bam\", \"s_epiblast.deduplicated.bam\", \"s_icm.deduplicated.bam\"]\n",
    "\n",
    "def methylation_extractor(input_file):\n",
    "    subprocess.call([\"bismark_methylation_extractor\", \"--gzip\",  \"--bedGraph\", \"--cytosine_report\", \"--genome_folder\", \"./mouse_chr\", input_file])\n",
    "    \n",
    "p1 = multiprocessing.Process(target=methylation_extractor, args=(input_file[0],))\n",
    "p2 = multiprocessing.Process(target=methylation_extractor, args=(input_file[1],))\n",
    "p3 = multiprocessing.Process(target=methylation_extractor, args=(input_file[2],))\n",
    "\n",
    "p1.start()\n",
    "p2.start()\n",
    "p3.start()\n",
    "\n",
    "p1.join()\n",
    "p2.join()\n",
    "p3.join()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "User specifified dedup report: SRR5836473_1_bismark_bt2_pe.deduplication_report.txt\n",
      "\n",
      "Writing Bismark HTML report to >> SRR5836473_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> ./cell8/SRR5836473_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report ./cell8/SRR5836473_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> SRR5836473_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report SRR5836473_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> s_8_cell.deduplicated_splitting_report.txt <\n",
      "Processing splitting report s_8_cell.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> s_8_cell.deduplicated.M-bias.txt <\n",
      "Processing M-bias report s_8_cell.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "User specifified dedup report: SRR5836475_1_bismark_bt2_pe.deduplication_report.txt\n",
      "\n",
      "Writing Bismark HTML report to >> SRR5836475_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> ./ICM/SRR5836475_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report ./ICM/SRR5836475_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> SRR5836475_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report SRR5836475_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> s_icm.deduplicated_splitting_report.txt <\n",
      "Processing splitting report s_icm.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> s_icm.deduplicated.M-bias.txt <\n",
      "Processing M-bias report s_icm.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "User specifified dedup report: SRR3824222_1_bismark_bt2_pe.deduplication_report.txt\n",
      "\n",
      "Writing Bismark HTML report to >> SRR3824222_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> ./epiblast/SRR3824222_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report ./epiblast/SRR3824222_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> SRR3824222_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report SRR3824222_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> s_epiblast.deduplicated_splitting_report.txt <\n",
      "Processing splitting report s_epiblast.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> s_epiblast.deduplicated.M-bias.txt <\n",
      "Processing M-bias report s_epiblast.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!bismark2report --alignment_report ./cell8/SRR5836473_1_bismark_bt2_PE_report.txt --mbias_report s_8_cell.deduplicated.M-bias.txt --dedup_report SRR5836473_1_bismark_bt2_pe.deduplication_report.txt --splitting_report s_8_cell.deduplicated_splitting_report.txt\n",
    "!bismark2report --alignment_report ./ICM/SRR5836475_1_bismark_bt2_PE_report.txt --mbias_report s_icm.deduplicated.M-bias.txt --dedup_report SRR5836475_1_bismark_bt2_pe.deduplication_report.txt --splitting_report s_icm.deduplicated_splitting_report.txt\n",
    "!bismark2report --alignment_report ./epiblast/SRR3824222_1_bismark_bt2_PE_report.txt --mbias_report s_epiblast.deduplicated.M-bias.txt --dedup_report SRR3824222_1_bismark_bt2_pe.deduplication_report.txt --splitting_report s_epiblast.deduplicated_splitting_report.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/lexa/anaconda3/envs/bio\n",
      "\n",
      "  added / updated specs:\n",
      "    - matplotlib=3.2\n",
      "\n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  ca-certificates    conda-forge::ca-certificates-2021.10.~ --> pkgs/main::ca-certificates-2021.10.26-h06a4308_2\n",
      "  certifi            conda-forge/linux-64::certifi-2016.9.~ --> pkgs/main/noarch::certifi-2020.6.20-pyhd3eb1b0_3\n",
      "\n",
      "The following packages will be SUPERSEDED by a higher-priority channel:\n",
      "\n",
      "  openssl            conda-forge::openssl-1.0.2u-h516909a_0 --> pkgs/main::openssl-1.0.2u-h7b6447c_0\n",
      "\n",
      "\n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/lexa/anaconda3/envs/bio\n",
      "\n",
      "  added / updated specs:\n",
      "    - intervaltree\n",
      "\n",
      "\n",
      "The following packages will be SUPERSEDED by a higher-priority channel:\n",
      "\n",
      "  ca-certificates    pkgs/main::ca-certificates-2021.10.26~ --> conda-forge::ca-certificates-2021.10.8-ha878542_0\n",
      "  certifi            pkgs/main/noarch::certifi-2020.6.20-p~ --> conda-forge/linux-64::certifi-2016.9.26-py36_0\n",
      "  openssl              pkgs/main::openssl-1.0.2u-h7b6447c_0 --> conda-forge::openssl-1.0.2u-h516909a_0\n",
      "\n",
      "\n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/lexa/anaconda3/envs/bio\n",
      "\n",
      "  added / updated specs:\n",
      "    - pybigwig\n",
      "\n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  ca-certificates    conda-forge::ca-certificates-2021.10.~ --> pkgs/main::ca-certificates-2021.10.26-h06a4308_2\n",
      "  certifi            conda-forge/linux-64::certifi-2016.9.~ --> pkgs/main/noarch::certifi-2020.6.20-pyhd3eb1b0_3\n",
      "\n",
      "\n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/lexa/anaconda3/envs/bio\n",
      "\n",
      "  added / updated specs:\n",
      "    - tqdm\n",
      "\n",
      "\n",
      "The following packages will be SUPERSEDED by a higher-priority channel:\n",
      "\n",
      "  ca-certificates    pkgs/main::ca-certificates-2021.10.26~ --> conda-forge::ca-certificates-2021.10.8-ha878542_0\n",
      "  certifi            pkgs/main/noarch::certifi-2020.6.20-p~ --> conda-forge/linux-64::certifi-2016.9.26-py36_0\n",
      "\n",
      "\n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/lexa/anaconda3/envs/bio\n",
      "\n",
      "  added / updated specs:\n",
      "    - pygenometracks\n",
      "\n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  ca-certificates    conda-forge::ca-certificates-2021.10.~ --> pkgs/main::ca-certificates-2021.10.26-h06a4308_2\n",
      "  certifi            conda-forge/linux-64::certifi-2016.9.~ --> pkgs/main/noarch::certifi-2020.6.20-pyhd3eb1b0_3\n",
      "\n",
      "\n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/lexa/anaconda3/envs/bio\n",
      "\n",
      "  added / updated specs:\n",
      "    - ipykernel\n",
      "\n",
      "\n",
      "The following packages will be SUPERSEDED by a higher-priority channel:\n",
      "\n",
      "  ca-certificates    pkgs/main::ca-certificates-2021.10.26~ --> conda-forge::ca-certificates-2021.10.8-ha878542_0\n",
      "  certifi            pkgs/main/noarch::certifi-2020.6.20-p~ --> conda-forge/linux-64::certifi-2016.9.26-py36_0\n",
      "\n",
      "\n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: failed with initial frozen solve. Retrying with flexible solve.\n",
      "Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.\n",
      "Collecting package metadata (repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/lexa/anaconda3/envs/bio\n",
      "\n",
      "  added / updated specs:\n",
      "    - ucsc-fetchchromsizes\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    mysql-connector-c-6.1.6    |                2         9.6 MB  bioconda\n",
      "    ucsc-fetchchromsizes-366   |       h5eb252a_0           5 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         9.6 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  mysql-connector-c  bioconda/linux-64::mysql-connector-c-6.1.6-2\n",
      "  ucsc-fetchchromsi~ bioconda/linux-64::ucsc-fetchchromsizes-366-h5eb252a_0\n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  ca-certificates    conda-forge::ca-certificates-2021.10.~ --> pkgs/main::ca-certificates-2021.10.26-h06a4308_2\n",
      "  certifi             conda-forge::certifi-2016.9.26-py36_0 --> pkgs/main::certifi-2021.5.30-py36h06a4308_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "ucsc-fetchchromsizes | 5 KB      | ##################################### | 100% \n",
      "mysql-connector-c-6. | 9.6 MB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/lexa/anaconda3/envs/bio\n",
      "\n",
      "  added / updated specs:\n",
      "    - ucsc-bedgraphtobigwig\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    ucsc-bedgraphtobigwig-357  |                1         973 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         973 KB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  ucsc-bedgraphtobi~ bioconda/linux-64::ucsc-bedgraphtobigwig-357-1\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "ucsc-bedgraphtobigwi | 973 KB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n"
     ]
    }
   ],
   "source": [
    "!conda install matplotlib=3.2 -y\n",
    "!conda install -c bioconda gffutils -y\n",
    "!conda install -c conda-forge intervaltree -y\n",
    "!conda install -c bioconda pybigwig -y\n",
    "!conda install -c bioconda hicmatrix -y\n",
    "!conda install -c bioconda pysam -y\n",
    "!conda install -c bioconda pybedtools -y\n",
    "!conda install -c conda-forge tqdm -y\n",
    "#Also you need python 3.6\n",
    "!conda install -c bioconda pygenometracks -y\n",
    "!conda install -c conda-forge ipykernel -y\n",
    "!conda install -c bioconda ucsc-fetchchromsizes -y\n",
    "!conda install -c bioconda ucsc-bedgraphtobigwig -y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: trying WGET /usr/bin/wget for database mm10\r\n",
      "url: http://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/mm10.chrom.sizes\r\n"
     ]
    }
   ],
   "source": [
    "!fetchChromSizes mm10 > m.chrom.sizes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "!gunzip s_8_cell.deduplicated.bedGraph.gz\n",
    "!gunzip s_epiblast.deduplicated.bedGraph.gz\n",
    "!gunzip s_icm.deduplicated.bedGraph.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedGraphToBigWig s_8_cell.deduplicated.bedGraph m.chrom.sizes s_8_cell_methylation.bw\n",
    "!bedGraphToBigWig s_epiblast.deduplicated.bedGraph m.chrom.sizes s_epiblast_methylation.bw\n",
    "!bedGraphToBigWig s_icm.deduplicated.bedGraph m.chrom.sizes s_icm_methylation.bw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "*****\n",
      "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
      "*****\n",
      "\n",
      "*****\n",
      "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
      "*****\n",
      "\n",
      "*****\n",
      "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
      "*****\n"
     ]
    }
   ],
   "source": [
    "!bedtools genomecov -bg -ibam s_8_cell.deduplicated.bam -g m.chrom.sizes > s_8_cell.cover.bedGraph\n",
    "!bedtools genomecov -bg -ibam s_epiblast.deduplicated.bam -g m.chrom.sizes > s_epiblast.cover.bedGraph\n",
    "!bedtools genomecov -bg -ibam s_icm.deduplicated.bam -g m.chrom.sizes > s_icm.cover.bedGraph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedGraphToBigWig s_8_cell.cover.bedGraph m.chrom.sizes s_8_cell_coverage.bw\n",
    "!bedGraphToBigWig s_epiblast.cover.bedGraph m.chrom.sizes s_epiblast_coverage.bw\n",
    "!bedGraphToBigWig s_icm.cover.bedGraph m.chrom.sizes s_icm_coverage.bw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:numexpr.utils:NumExpr defaulting to 8 threads.\n",
      "Adding bigwig file: s_8_cell_methylation.bw\n",
      "Adding bigwig file: s_8_cell_coverage.bw\n",
      "INFO:numexpr.utils:NumExpr defaulting to 8 threads.\n",
      "Adding bigwig file: s_epiblast_methylation.bw\n",
      "Adding bigwig file: s_epiblast_coverage.bw\n",
      "INFO:numexpr.utils:NumExpr defaulting to 8 threads.\n",
      "Adding bigwig file: s_icm_methylation.bw\n",
      "Adding bigwig file: s_icm_coverage.bw\n"
     ]
    }
   ],
   "source": [
    "!make_tracks_file --trackFiles s_8_cell_methylation.bw s_8_cell_coverage.bw -o s_8_cell_tracks.ini\n",
    "!make_tracks_file --trackFiles s_epiblast_methylation.bw s_epiblast_coverage.bw -o s_epiblast_tracks.ini\n",
    "!make_tracks_file --trackFiles s_icm_methylation.bw s_icm_coverage.bw -o s_icm_tracks.ini"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:numexpr.utils:NumExpr defaulting to 8 threads.\n",
      "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
      "INFO:pygenometracks.tracksClass:0.0021605491638183594\n",
      "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 6.0. Dpi is set to 72\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 1. [x-axis]\n",
      "INFO:pygenometracks.tracksClass:plotting 2. [spacer]\n",
      "INFO:pygenometracks.tracksClass:plotting 3. [s_8_cell_methylation]\n",
      "INFO:pygenometracks.tracksClass:plotting 4. [s_8_cell_coverage]\n",
      "INFO:numexpr.utils:NumExpr defaulting to 8 threads.\n",
      "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
      "INFO:pygenometracks.tracksClass:0.0020961761474609375\n",
      "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 6.0. Dpi is set to 72\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 1. [x-axis]\n",
      "INFO:pygenometracks.tracksClass:plotting 2. [spacer]\n",
      "INFO:pygenometracks.tracksClass:plotting 3. [s_epiblast_methylation]\n",
      "INFO:pygenometracks.tracksClass:plotting 4. [s_epiblast_coverage]\n",
      "INFO:numexpr.utils:NumExpr defaulting to 8 threads.\n",
      "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
      "INFO:pygenometracks.tracksClass:0.002183675765991211\n",
      "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 6.0. Dpi is set to 72\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 1. [x-axis]\n",
      "INFO:pygenometracks.tracksClass:plotting 2. [spacer]\n",
      "INFO:pygenometracks.tracksClass:plotting 3. [s_icm_methylation]\n",
      "INFO:pygenometracks.tracksClass:plotting 4. [s_icm_coverage]\n"
     ]
    }
   ],
   "source": [
    "!pyGenomeTracks --tracks s_8_cell_tracks.ini --region chr11:0-122082543 -o s_8_cell_image.png\n",
    "!pyGenomeTracks --tracks s_epiblast_tracks.ini --region chr11:0-122082543 -o s_epiblast_image.png\n",
    "!pyGenomeTracks --tracks s_icm_tracks.ini --region chr11:0-122082543 -o s_icm_image.png"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAEWCAYAAABrDZDcAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deZxcVZn/8c83nX0PJCzZA0QQ+LEZUFxRZB0YUGEAd1EZFAScYUYQZ9SR+Snu4qAxAuKIsoygRgQiA7gBSoIGDEsgJoFskA5kT8j6zB/3NFQX1d2VTt3qrr7f9+tVr667Pufeqq7n3nPPPVcRgZmZFVevri6AmZl1LScCM7OCcyIwMys4JwIzs4JzIjAzKzgnAjOzgnMisLqSFJL2Se+vk3R5lcsdJWlxJ2N2etkK63qpzJLeJGluLda7k2X6nKTr25j2QUl/qFGcienz612L9XWyDDXbHnuZE0HBKXO5pCWSVkv6jaQDurpcjSAifh8R++7seiQtlPT2WpSp3iQdI+leSWslPS9ptqRPSerf1WWz6jkR2OnA2cCbgF2AB4AfdWmJrCFIOh34KfATYEJE7AqcAYwFxrWxTJedTVjbnAh6oHREtiQdpc2VdHQ7s08C/hAR8yNiG3A9sH8VMcZJulVSczoS/K+SaWdLelzSSkkzJE3oxDYMSNUwKyU9BhxeNn20pFtS/AWSLtiBZV+qnkrDpdU9R0laLOnTklako/X3tFHGVlVObe0TSXtLuieNWyHpx5KGp2k/AsYDv5S0TtK/pvGvk3S/pFWSHpZ0VEmcSZJ+mz7fu4CRHe9OfTud8T1R+n2QNEzSNZKWpe/M5ZKa0rQmSV9NZZ4P/F3pCoGvA/8REd+PiBcAImJuRHwiIp5K831O0k8lXS9pDfBBSUdIeiBt2zJJ/yWpb9nnc4Gk+Sn2VyS1+q1K5VqZPvsTOth+64ATQQ8jaV/gfODwiBgCHAcsbGeRG4F9JL1KUh/gA8CdHcRoAm4DngYmAmPSepB0KvBp4J3AKOD3wA2d2JTPAnun13GpXC3xewG/BB5OsY8GLpJ0XEfLVmkPsh/XMWnZaWm/tqm9fQII+CIwGng12dHy5wAi4n3AM8DJETE4Ir4saQzwK+BysrO0i4FbJI1K6/sJ8FAq4xeq2L7XAvPT/J8FbpW0S5r2Q2ArsA9wKHAs8JE07aPASWn8FOC0knXuS3bkf0sHsQFOITtzGA78GNgGfDKV50iyz+/jZcu8I8U8LC1/dtn2zE3Lfxm4JiUm66yI8KsHvcj+oZcDbwf6VDF/X+BbQJD9ICwAJnWwzJFAM9C7wrQ7gA+XDPcCNpBVHZDi7JPeXwdc3kaM+cDxJcPnAIvT+9cCz5TNfynwg46WLS9DeTmAo9J+GFQy/Wbg39qYt6VMbe6TCtt2KvCXkuGFwNtLhj8F/KhsmRlkP/jjK5TvJ8D1bcT6ILAUUMm4B4H3AbsDm4ABJdPOAu5N7+8Bzi2Zdmzad72BN6b3/Uum3wisSp/3+9K4zwG/62B/XAT8rOzzKf38Pg7cXbI980qmDUzz79HV/3uN/HJ9XQ8TEfMkXUT2D3iApBnAP0XE0jYW+SxZ1ck44FngvcA9kg6IiA1tLDMOeDoitlaYNgH4lqSvlYwT2RHy0zuwKaOBRSXDpctOAEZLWlUyrons7KOjZauxMiLWly0/uoNl2twnknYDriS7DjOELDmubGddE4DTJZ1cMq4PcG8qR6XyVayTT5ZE+tUsmX90itMHWFZyQN2Ll/dde/vx+fR3T7KDByLiTABlrXqaSuYtXQeSXkVWrTSF7Ie8N9kZDm0sU77/n215ExEbUtkHY53mqqEeKCJ+EhFvJPtHD+CKdmY/GLgpIhZHxNaIuA4YQfvXCRYB49u48LcI+MeIGF7yGhAR9+/gZiyj9Y/b+LIYC8piDImIE6tYFrIj1oElw3uUTR8haVDZ8m0l0tIytbVPvkj2ORwUEUPJkm1pVUZ5F8CLyM4ISrdvUER8iWzbKpWvPWPKqk5atmcR2RnByJI4QyOipdVYe/vxCWAJWRVgR8q377tp+clpf3ya1vuDCnE72v+2E5wIehhJ+0p6m6R+wIvARrI62bbMJDv63F1SL0nvIztKnNfOMg+S/Uh8SdIgSf0lvSFNmwpcqtQENV2MPL0Tm3JzWs8ISWOBT5TFX6PsoviAdFHzQEmHV7EswGzg3Wm544G3VIj/eUl9Jb2JrJ78fzoob3v7ZAiwDliV6v//pWzZ54C9SoavB06WdFwqY/90YXpsRDwNzCop3xuBk2nfbsAFkvqkz+LVwO0RsQz4NfA1SUPT57+3pJb9cXNabqykEcAlLStMZxj/DHxW0kfTvpakyWRVTu0ZAqwB1knaD/hYhXn+Ja1zHHAhcFMH67Sd4ETQ8/QDvgSsIDuF3o3siKstV5BddJ1NVr/7SeBdEbGqrQUia110Mtn1iGeAxWTNBomIn6V13phaicwBOmzVIWm8slYzLUednyerElhA9mP1UpPWkviHpOkrgKuBYR0tm1yYll8FvAf4edn0Z8mqbpaSXdw8NyKeaK/87e2TVJ7DgNVkF4FvLVv8i8BnUiuaiyNiEdkF0k+TXXdYRJY8Wv5f3012neQFsqq9/y5dWdqPbyoZ9SdgMtl++k/gtIhoqdp5P9l1osfSNv+UrLoH4Ptk1yYeBv5cXu6IuAn4B7IznEVp/TcD02g/cV6ctmFtilHpR/4XZNVFs8n22TXtrM92klpXHZoVm7JmmtdHxNiuLktRSQqyaqP2zkqthnxGYGZWcE4EBSDpjlRdUP5qs8qopKqm0quji5Nm1kBcNWRmVnA+IzAzK7iGu6Fs5MiRMXHixK4uhplZQ3nooYdWRMSoStMaLhFMnDiRWbNmdXUxzMwaiqQ277B31ZCZWcE5EZiZFZwTgZlZwTkRmJkVnBOBmVnBORGYmRWcE4GZWcEVJhHMfXYtX//1XFas29TVRTEz61YKkwjmLV/HlffM44X1m7u6KGZm3UphEoGZmVXmRGBmVnBOBGZmBedEYGZWcIVLBH4Oj5lZa4VJBFJXl8DMrHsqTCIwM7PKnAjMzArOicDMrOCcCMzMCq5wiSBwsyEzs1KFSQRuNGRmVllhEoGZmVXmRGBmVnBOBGZmBedEYGZWcE4EZmYFV7hE4E7nzMxayzURSDpe0lxJ8yRd0s58h0vaJum0/MqS15rNzBpbbolAUhNwFXACsD9wlqT925jvCmBGXmUxM7O25XlGcAQwLyLmR8Rm4EbglArzfQK4BVieY1nMzKwNeSaCMcCikuHFadxLJI0B3gFMbW9Fks6RNEvSrObm5poX1MysyPJMBJVq5csv1X4T+FREbGtvRRExLSKmRMSUUaNG1ayAZmYGvXNc92JgXMnwWGBp2TxTgBuVXckdCZwoaWtE/DyvQrnVkJlZa3kmgpnAZEmTgCXAmcC7S2eIiEkt7yVdB9yWXxJwsyEzs0pySwQRsVXS+WStgZqAayPiUUnnpuntXhcwM7P6yPOMgIi4Hbi9bFzFBBARH8yzLGZmVlnh7iw2M7PWnAjMzAqucInAj6o0M2utMInAfQ2ZmVVWmERgZmaVORGYmRWcE4GZWcE5EZiZFVzhEoH7GjIza60wicCNhszMKitMIjAzs8qcCMzMCs6JwMys4JwIzMwKzonAzKzgCpMI5M6GzMwqKkwiMDOzypwIzMwKzonAzKzgnAjMzAqucInAfQ2ZmbVWuERgZmatFSYRuPGomVllhUkEZmZWmROBmVnBORGYmRWcE4GZWcEVLhEEbj9qZlaqMInAfc6ZmVVWmERgZmaVORGYmRWcE4GZWcE5EZiZFVzhEoE7nTMza60wicCthszMKss1EUg6XtJcSfMkXVJh+imSHpE0W9IsSW/MszxmZvZKvfNasaQm4CrgGGAxMFPS9Ih4rGS2u4HpERGSDgJuBvbLq0xmZvZKeZ4RHAHMi4j5EbEZuBE4pXSGiFgX8VKt/SDwbb9mZvWWZyIYAywqGV6cxrUi6R2SngB+BZxdaUWSzklVR7Oam5tzKayZWVHlmQgqXZ59xRF/RPwsIvYDTgW+UGlFETEtIqZExJRRo0btVKF8ymFm1lqeiWAxMK5keCywtK2ZI+J3wN6SRuZRGPkZZWZmFeWZCGYCkyVNktQXOBOYXjqDpH2krGGnpMOAvsDzOZbJzMzK5NZqKCK2SjofmAE0AddGxKOSzk3TpwLvAt4vaQuwETij5OKxmZnVQW6JACAibgduLxs3teT9FcAVeZbBzMzaV5g7i83MrLLCJQLXPJmZtVacROBGQ2ZmFRUnEZiZWUVOBGZmBddhIkhdO5wnaUQ9CmRmZvVVzRnBmcBost5Db5R0XMtNYGZm1vg6TAQRMS8iLgNeBfwEuBZ4RtLnJe2SdwFrzW2GzMxaq+oaQXpWwNeArwC3AKcBa4B78iuamZnVQ4d3Fkt6CFgFXANcEhGb0qQ/SXpDnoWrJddlmZlVVk0XE6dHxPxKEyLinTUuj5mZ1Vk1VUMfkTS8ZUDSCEmX51gmMzOro2oSwQkRsaplICJWAifmVyQzM6unahJBk6R+LQOSBgD92pm/W3NXQ2bWiC644S/8YvaSXNZdzTWC64G7Jf2ArPXl2cAPcymNmZlVdOecZxkzYgCn5LDuDhNBRHxZ0l+Bo8ka33whImbkUJZc+R44M7PKqnowTUTcAdyRc1nMzKwNkePtsNX0NfROSU9JWi1pjaS1ktbkViIzM6sor3qNas4IvgycHBGP51QGMzPrQtW0GnrOScDMrOeq5oxglqSbgJ8DLd1LEBG35laqXLn9qJk1njybvleTCIYCG4BjS8YF0FCJwG2GzKzR5dX4sZrmox/KJ7SZmXUH1bQaepWkuyXNScMHSfpM/kUzM7MWeVZqV3Ox+PvApcAWgIh4hOypZWZmVkfKqZK7mkQwMCIeLBu3NY/CmJlZ/VWTCFZI2pt0ZiLpNGBZrqXKkTudMzNrrZpWQ+cB04D9JC0BFgDvzbVUOXBXQ2bWyCLHo9hqWg3NB94uaRDQKyLW5lYaMzNrU5c1H5X072XDAETEf+RTJDMzq6dqqobWl7zvD5wEuMsJM7M6yvPyZjVVQ18rHZb0VWB6biUyM7OK8rrUWU2roXIDgb1qXZB6caMhM7PWqrlG8Fde/v1sAkYBDXd9IK8bMczMGl011whOKnm/laxbat9QZmZWR3neA1VN1dDaktdGYKikXVpe7S0o6XhJcyXNk3RJhenvkfRIet0v6eBObYWZWRHk1H60mjOCPwPjgJVk1yqGA8+kaUEb1wskNQFXAccAi4GZkqZHxGMlsy0A3hIRKyWdQHbj2ms7syFmZtY51ZwR3En2qMqREbErWVXRrRExKSLau2h8BDAvIuZHxGbgRuCU0hki4v6IWJkG/wiM3fFNMDOznVFNIjg8Im5vGYiIO4C3VLHcGGBRyfDiNK4tHwbuqDRB0jmSZkma1dzcXEXotrmvITNrVF3ZfHSFpM9ImihpgqTLgOerWK5SmSv+DEt6K1ki+FSl6RExLSKmRMSUUaNGVRHazMyqVU0iOIusyejP0mtUGteRxWTXFlqMBZaWzyTpIOBq4JSIqCbBdIo7nTMzq6yaO4tfAC6UNDgi1u3AumcCkyVNApaQPczm3aUzSBpP9uzj90XEkzuwbjOzwsiz51Go7lGVr5f0GPBYGj5Y0nc6Wi7da3A+MIOsb6KbI+JRSedKOjfN9u/ArsB3JM2WNKuzG2Jm1tN1We+jwDeA40j9C0XEw5LeXM3K00Xm28vGTS15/xHgI1WX1szMaq6qvoYiYlHZqG05lKUu8j7FMjOrtbx/tqo5I1gk6fVASOoLXIC7oTYz6zGqOSM4l+xxlWPIWgIdkoYbihsNmVmjy6vzzHbPCFI3Ed+MiPfkEt3MzLpcu2cEEbENGJWqhMzMrAvkfWWzmmsEC4H7JE2n5LGVEfH1vAplZmavlFfz0TbPCCT9KL09A7gtzTuk5GVmZj1Ae2cEr5E0gazL6W/XqTy5c+NRM2s0eTd7by8RTCXrgnoSUHrHr2jnOQTdlpsNmZlV1GbVUERcGRGvBn4QEXuVvDp6DoGZmeWgy7qhjoiP5RTbzMy6gaq6mDAzs66T97VNJwIzswZR9+ajPZX7nDMza60wiSCvPjrMzPKW9wFsYRKBmZlV5kRgZtYglNNFAicCM7OCcyIwM+vmIucGpIVLBHnvUDOzRlOYRJBX+1szs0ZXmERgZtao3HzUzMxy5URgZtYg3MWEmZnloniJwI2GzMxaKV4iMDNrUHn1mVaYRODWo2ZmlRUmEZjV2rbtwcRLfsVV987r6qJYD+fmo2bd1JZt2wG48u6nurgkZjvHicDMrEG4+WiNuNGQ1Zq/U9boCpcIzMwajXsfrZG8Huhg5m+W1Ute37XCJAIzM6ss10Qg6XhJcyXNk3RJhen7SXpA0iZJF+dZFjOzRpV389Heea1YUhNwFXAMsBiYKWl6RDxWMtsLwAXAqXmVwyxvvlhsjS7PM4IjgHkRMT8iNgM3AqeUzhARyyNiJrAlx3K0kndmteLwd8nqrRGbj44BFpUML07jdpikcyTNkjSrubm5JoUz21ktLTl8sdgaXZ6JoNL/R6eOoSJiWkRMiYgpo0aN6lxh/N9qZg0q75PPPBPBYmBcyfBYYGmO8czqylVDVm+N2PvoTGCypEmS+gJnAtNzjGfWJZwPrNHl1mooIrZKOh+YATQB10bEo5LOTdOnStoDmAUMBbZLugjYPyLW5FUus1pxArB6iZxPP3NLBAARcTtwe9m4qSXvnyWrMjJrOC3/nL78ZI2ucHcW591nh5lZXhqx+Wi34qM2qzUfUlhPUZhEYJYXJwTLWyM3HzXr0dx81HoKJwKzneRqR2t0TgRmneUzAqsTP7y+xnw6b7XiFmjWUxQmEbivITNrdHk9abEwicCs1nx2aT2FE4GZWXfnawRm3VOU/TXLmx9eb9bN5N0RmFm9FC4R+F/Xas3tECxvebdQK1wiMKsVH1RYT1GgRODjNjNrbO591KybablE4DMDa3ROBPaShxetYuIlv+KBvz3f1UVpCL6z2OrFXUxY3dyfEsBvnlzexSVpLK50tHpx89EacZM/qxl/layHKFwiMLPaiAiO+fpv+cXsJV1dlB7PD6apEXc6Z7VW9DuLt20Pnlq+jk/eNLuri2I7qTCJwKzWil7LWPDN7xLufdSsmyrqyWbRE2FP4kRg1klFbz663ZmgbvJu5FK4ROCvrlltOA/Un+8sNutmin5nccsZUVG3vycpTCIoaj2uWV62OwPUjZuPmnVTLf+cRT3IaLlGUNTt70mcCMw6qeUCXlEPjGN7V5egeNzFRIHdO3c5c59d29XFMGvF1wh6jt5dXYC6a8Bv7Yd+MBOAhV/6uy4uiZUqeqsZXyOoH/c+ambdUldcI7j01r9y/7wVdYzYzfjO4p2T163ZZkX9ZtX7hrKI4IYHn+HdV/+prnGLoDCJwKzWin4fQb03fGuB66L88PoeaubCF2heu6mri9EtLF21kRsffKari9GjLF21kSefy7eBwfY6J8It29xMKS+5JgJJx0uaK2mepEsqTJekK9P0RyQdlmd5upPTpz7AaVPv7+pidAvvv/ZBLrn1r6zeuKWri7JDunNfQ6//0j0c+43f5Rqj3lVDW7bVJt7aF7dw1rQ/snDF+pqsr54arvmopCbgKuAEYH/gLEn7l812AjA5vc4BvptXeepl5frNbE+HSr98eCn//cDCV8yzeWt2ZPP08xtqGnv1hi0sX/NiTdfZWRs2b2VrlUdwz63Oyrxp67YdirF8zYvMW97+Ue/2BqxOaKuDsY2bt/FUjY/yN27esX1eqhY31C16YUPV35Nq5+vI3Y8v54H5z/O1u56sav7fPtnMnCWr25zeaAcwlSivXu0kHQl8LiKOS8OXAkTEF0vm+R7wm4i4IQ3PBY6KiGVtrXfKlCkxa9asHS7P7EWrOPWq+xg9rD+D+tW21eyKdZsYOqAPvSQWpKOMybsN5qnl6156X2pbBPObs/n2SdM2bd3Gpi3bGTqgzyvWP6+N9ZSrFG/lhi0M6NuL/r2bOtyOluWriVXtuqpZT8u8E3cdSJ+m6o9NOoqxYt0mNm/dzujhA4Dsh2v5mhcZObgfTb12/thqy7btLEzJfGf2V/lntGjlBoYP6MuQ/q2/py3bO2nkIHp3UP5q9v+KdZtYuWELY4YPYGDfLPaLW7exbVswsIr/ka07uP0rN2xmcL/eL33Gm7Zu55kXql9+6/Zo9f/V3nwr1m1i96H9KyapJas2siElwB35flaa9/n1m3lh/WaGD+zDqMH9gOx7sXztJvYc1p9eNWqk0rLtl596IO993YROrUPSQxExpdK0PO8jGAMsKhleDLy2innGAK0SgaRzyM4YGD9+fKcKs+/uQzhjyjjWbqp99t51cF/6NPViSP/eLFixnsPGD2ePYf1ZvXELqzduYfLur/wCzW9ezwGjhzJx10FAVs3QvHYTuw3p/4p55y1fx14jB1VcT6k1L25h+dpNreZb++JWtkcwrEKCKbf3qMHc+eizHLv/7vRu2rkv8FNVlhlg6IA+PPT0SvYfPXSHYmzYvI0lqza2GWPP4QN4fNmaVtOH9O/NroP60rd3bU6GFz6/gUPHD2fPYa/83KpV/hkN6NuEgDEjBrSab0DfJh5ZvJpX7zmkw3Vu2LyN5WtfbHf/7zl8AL97spkDxwx9KTFu354d4e4yqG9VZV/4/AYOHjecMcM73v5VG7bQp6kXg/q9fFDyzAsbmDJhBLsN7VdVvAUr1vP/xgxj3C4D2p1vfnMvJo0cVLG15T67DeaOOc/ytv12o3+fjr8Hf2texx5D+1fclxO2Bv/7+HNMmTCi1XdKy9Yyebch9KphnctBY4fx5smjarfCEnkmgkq/JOWnH9XMQ0RMA6ZBdkbQmcIM6NvEFacd1JlFu9xVXV0Aszb4u9kz5HmxeDEwrmR4LLC0E/OYmVmO8kwEM4HJkiZJ6gucCUwvm2c68P7Ueuh1wOr2rg+YmVnt5VY1FBFbJZ0PzACagGsj4lFJ56bpU4HbgROBecAG4EN5lcfMzCrLtdO5iLid7Me+dNzUkvcBnJdnGczMrH2+s9jMrOCcCMzMCs6JwMys4JwIzMwKLrcuJvIiqRl4upOLjwS66qkWXRXb21yM2N7mnh93Z2NPiIiKtyY3XCLYGZJmtdXXRk+N7W0uRmxvc8+Pm2dsVw2ZmRWcE4GZWcEVLRFMK2Bsb3MxYnube37c3GIX6hqBmZm9UtHOCMzMrIwTgZlZwRUmEUg6XtJcSfMkXZJzrGslLZc0p2TcLpLukvRU+jsih7jjJN0r6XFJj0q6sB6xJfWX9KCkh1Pcz9cjbkn8Jkl/kXRbneMulPRXSbMlzapXbEnDJf1U0hPpsz6yTnH3Tdva8loj6aI6xf5k+m7NkXRD+s7V63O+MMV9VNJFaVzNY+/o74akS9Pv2VxJx+1M7EIkAklNZA9TOgHYHzhL0v45hrwOOL5s3CXA3RExGbg7DdfaVuCfI+LVwOuA89J25h17E/C2iDgYOAQ4Pj1foh7bDHAh8HjJcL3iArw1Ig4padtdj9jfAu6MiP2Ag8m2Pfe4ETE3beshwGvIuo7/Wd6xJY0BLgCmRMSBZN3an5l33BT7QOCjwBFk+/okSZNzin0dVf5upP/rM4ED0jLfSb9znRMRPf4FHAnMKBm+FLg055gTgTklw3OBPdP7PYG5ddjuXwDH1DM2MBD4M9nzqXOPS/ZUu7uBtwG31XNfAwuBkWXjco0NDAUWkBp6dNX3CzgWuK9O29zybPNdyLrOvy3Fr8f363Tg6pLhfwP+Na/Y1f5ulP+GkT335cjOxi3EGQEvf5FaLE7j6mn3SE9fS393yzOYpInAocCf6hE7Vc/MBpYDd0VEXeIC3yT7x9xeMq5e+zqAX0t6SNI5dYq9F9AM/CBVh10taVAd4pY7E7ghvc81dkQsAb4KPAMsI3uS4a/zjpvMAd4saVdJA8kepDWuTrFpJ05Nf9OKkghUYVyPbTcraTBwC3BRRKypR8yI2BZZlcFY4Ih0Sp0rSScByyPiobxjteENEXEYWZXjeZLeXIeYvYHDgO9GxKHAevKt+noFZY+e/Xvgf+oUbwRwCjAJGA0MkvTeesSOiMeBK4C7gDuBh8mqYLtaTX/TipIIFpNl8RZjgaV1LsNzkvYESH+X5xFEUh+yJPDjiLi1nrEBImIV8Buyesu8474B+HtJC4EbgbdJur4OcQGIiKXp73KyuvIj6hB7MbA4nXEB/JQsMdTtMyZLfH+OiOfScN6x3w4siIjmiNgC3Aq8vg5xAYiIayLisIh4M/AC8FS9YrcTp6a/aUVJBDOByZImpaOZM4HpdS7DdOAD6f0HyOrva0qSgGuAxyPi6/WKLWmUpOHp/QCyf9wn8o4bEZdGxNiImEj2md4TEe/NOy6ApEGShrS8J6uznpN37Ih4Flgkad806mjgsbzjljmLl6uFqEPsZ4DXSRqYvuNHk10gr8s2S9ot/R0PvJNs2+u1v9uKMx04U1I/SZOAycCDnY5S64sr3fVFVrf3JPA34LKcY91AVpe5hSxzfxjYleyi5lPp7y45xH0j2enhI8Ds9Dox79jAQcBfUtw5wL+n8blvc0kZjuLli8X12Nd7kVUTPAw82vKdqlPsQ4BZaX//HBhRr31N1hjgeWBYybh6bPPnyQ4u5gA/AvrVcZt/T5ZsHwaOzmubd/R3A7gs/Z7NBU7YmdjuYsLMrOCKUjVkZmZtcCIwMys4JwIzs4JzIjAzKzgnAjOzgnMiMKuD1EvnwK4uh1klbj5q1gZJvSOiJt0JpLufp0TEilqsz6yWfEZgPZqkianf/h9KeiT14z9Q0msk/TZ1GDej5Db+30j6/5J+C1wo6XBJ9yt71sKDkoakDva+ImlmWuc/pmWPSsu3PCvgx8pcQNZHzr2S7k3zflfSLJU8vyGNPzEt+wdJV+rlZywMUtZf/czU2dwpdd+Z1mP17uoCmNXBvsCHI+I+SdcC5wHvAE6JiGZJZwD/CZyd5h8eEW9J3ZE8ARhbNhMAAAIoSURBVJwRETMlDQU2kt3xuToiDpfUD7hP0q/TsoeS9RG/FLiPrGO6KyX9E9mzC1rOCC6LiBdSH/J3SzqI7M737wFvjogFkkq7cbiMrAuNs1N3Hg9K+t+IWJ/D/rKCcSKwIlgUEfel99cDnwYOBO7Kuq6hiezW/hY3pb/7AssiYiZApJ5cJR0LHCTptDTfMLK+XjYDD0bE4jTfbLL+5f9QoUz/kLqu7k3Wz/z+ZGfo8yNiQZrnBqCle+tjyTrYuzgN9wfG0/qBPGad4kRgRVB+IWwt8GhEHNnG/C1H2aqwbMv4T0TEjFYjpaPIntbWYhsV/sdSJ2EXA4dHxEpJ15H9sFfqWrg05rsiYm4785h1iq8RWBGMl9Tyo38W8EdgVMs4SX0kHVBhuSeA0ZIOT/MNkdSb7GlQH0tdfiPpVakH0vasBYak90PJks1qSbuTdevcEm8vZQ8VAjijZPkZwCdS75tIOrTjzTarjs8IrAgeBz4g6XtkvTh+m+yH9UpJw8j+D75J1ovoSyJic7p+8O3UvfZGsi62ryar8vlz+mFuBk7toAzTgDskLYuIt0r6S4o3n+xaAhGxUdLHgTslraB1t8JfSGV8JMVcCJzUmZ1hVs7NR61HS0fXt0X20PNuT9LgiFiXfuyvAp6KiG90dbmsZ3PVkFn38tF0kflRsovQ3+vi8lgB+IzAzKzgfEZgZlZwTgRmZgXnRGBmVnBOBGZmBedEYGZWcP8HZmtZQxFaSZ4AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAEWCAYAAABrDZDcAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3de5wddX3/8dc7CSTkSiBLhFxIwAAGRcSAYilSAQVqi1UpQWlVtEgFlaq/nyBqsWp/3kUtNlJEqwgRFTVFIFDkIhchAQEJEHIhkJCQbO73yyaf3x/zPcnsydndk2TnbHbn/Xw8zmPPzHxnvp+ZMzufM9+Z8x1FBGZmVl69ujoAMzPrWk4EZmYl50RgZlZyTgRmZiXnRGBmVnJOBGZmJedEYO2S9BlJ16b3YySFpD5tlL1S0vWNjXDP6u3MmCXNk3Raer99u3UlSfdI+lAb034s6UudVM/7Jd3fGcvagxg6bX3KxonA2hUR/x4RNQ8kRZB0iqQFjaqvKJ2x3TpKvHszZS6R9KSk9ZJeTklpYlfHZjtzIjCzInwXuBT4JHAgMAL4LHBGrcIpcfh41EW84bspSZ+W9JKkNZJmSjq1nbK9JF0maY6kZZJuknRAmlb51nmhpIWSFkn6ZG7eWk0nF9QqW6PeX6Rvgqsk3Sfp6Ny0syQ9neJ/SdKnJA0AbgMOkbQ2vQ6psdyxku5N894JDKua/kZJD0paKekJSafUM2+ts5Gq5p4rJf1S0s/T/I9Jem0b695qu0k6KRfTfEnvT+P/WtKfJK1O46/MLea+9Hdl2hYnpnkukPSMpBWSpko6NFfP6ZKeTdv8PwDVii9nmKQ70/rcW7Wso9K05Wkf+/vctAMlTUlxPwIcnpt2BPARYGJE3BkRGyJia0TcHxHvz5W7R9KXJT0ArAcOk/SBtG5rJM2V9OHqz0dZs9vS9Nm8t2p9hkr6XZr/YUmHYx2LCL+62Qs4EpgPHJKGxwCHt1P+UuCPwEigL/AD4MbcvAHcCAwAXgM0A6el6VcC1+9q2TR8ATAo1XkV8Hhu2iLgL9P7ocBx6f0pwIIO1v8h4FtpuScDa3IxjgCWAWeRfdE5PQ031THvTnUD86rWbwvwbmAf4FPA88A+bZStLHd0que8NN+BwLG5Ol+TYj0GWAy8o2p798nF8w5gNvAqoA/Zt+wH07RhwOpcfP8CtAAfamM7/jjFdXLaHt8B7k/TBpDtYx9I9RwHLAWOTtMnAzelcq8GXsrNexEwr479+B7gReDoVMc+wF+TJRUBbyZLEPl9oyX3+b0ZWAccmVuf5cAJaXk/AyZ39f9rd3h1eQB+7caHBq8ElgCnVQ5CHZR/Bjg1N3xwOqD1yR1sjspN/xrww/Q+f0Cru2yNGPZP8w5Jwy8CHwYGV5U7hXYSAdlBtQUYkBt3Qy7GTwM/rZpnKvC+OubdqW52Prj/MTetF60TWnXZynIvB35d52d7FfDtqu2dTwS3AR+simE9cCjwj1XxCVhA+4lgcm54ILAVGAWcC/yhqvwPgH8Feqf9J78f/Ds7EsFn83GkcQuAlcBG4NA07h7g3zrYHr8BPp77fKo/v5uAz+XW59rctLOAZ7vyf7W7vNw01A1FxGyyb/lXAkskTa7VhJJzKPDr1CyxkiwxbAWG58rMz71/AWhveR2WldRb0ldSc9RqsoMk7GiKeRfZP+oLqUnixHbqyzsEWBER66piqDgUOKeyrml9TyJLfh3NW4/t6x4R28gOcO1tK8gOrHNqTZD0Bkl3S2qWtIrs2/SwWmWTQ4Hv5NZtOdkBf0SKIx9f0Pqz6mh91qblHZLqeUPVdnwv8AqgiexLRPV+ULGMbHtvFxEj03r1pXVzVav4JJ0p6Y+pOWol2T6S3x61Pr/89n859349WXKzDjgRdFMRcUNEnET2DxvAV9spPh84MyL2z736RcRLuTKjcu9HAwvbWV49Zd8DnE121jKE7NstpINAREyLiLOBg8i+9d1UWbV26oXsG/jQdD0hH0PFfLIzgvy6DoiIr9Qx7zqgf2VAUm+yg17eqNz0XmTNbe1tq0pMbbVV3wBMAUZFxBBgEjsOlLW2xXzgw1Xrt19EPJjWLx+faP1Z1ZIvPxA4IK3PfODeqnoGRsQ/kzUHtrDzflDxe2CkpAkd1N1qHSX1BX4FfAMYHhH7A7fSOnHU+vw62v7WASeCbkjSkZLekv5xNgIbyL7ht2US8OXKhUBJTZLOrirzOUn9lV3Q/QDw83aWV0/ZQcAmsm+H/cmaDirx7yvpvZKGRMQWsnbtSvyLgQMlDalVcUS8AEwHvpCWcxLwN7ki1wN/I+lt6aykX7rIOLKOeZ8D+qULuPuQNXH0rQrh9ZLeqeyWzkvTOv6xnW0FWVv1aZL+XlKfdKH12Nx2Wh4RGyWdQJZAK5qBbcBhuXGTgMvTtkfSEEnnpGm/A47Oxfcxsm/w7TlL2YXsfYEvAg9HxHzgFuAISf8gaZ/0Ol7SqyJiK3AzcGXaD8aTNb0BEBEzyZqRJqeL1/ulpPqmDmLZl2x7NwMtks4E3lqjXOXz+0vg7cAvOliudcCJoHvqC3yF7OLdy2Tfqj/TTvnvkH3rvEPSGrID1xuqytxLdhHyLuAbEXFHO8urp+xPyE7bXwKeZueD5T8A81Kz0UXA+QAR8SzZxei5qUnikJQ0ZuTmfU+KfzlZm/VPKhPSQexssu3RTPbN9v+wY19vb95VZHe7XJviXkfW9JP3W7L28xVpHd6ZklmbIuJFsiaOT6Z6Hwcqdxt9BPi39Ll8nh1nRkTEeuDLwANpW7wxIn5NdvY3OW27p4AzU/mlwDlk+8YyYBzwQGV5kv5S0tqq8G5I22E58Hqy5h8iYg3ZQXgi2Tful1O9lcR4CVmzy8tkbfM/qlruxWS3kH4rLXsBWaI5l+z6UK3ttIYsed1Etn3fQ7bf5r2cpi0kS7AXpX3G9oCyZkQrK0lj2HHnS0vXRrN3U3Zr5ysj4vyujqWMlN0GfH263mCdyGcEZmYl50TQQ0i6TTt+hJV/tddkZGbmpiEzs7LzGYGZWcl1u14Nhw0bFmPGjOnqMMzMupVHH310aURU/y4G6IaJYMyYMUyfPr2rwzAz61YktfkrejcNmZmVnBOBmVnJORGYmZWcE4GZWck5EZiZlZwTgZlZyTkRmJmVnBOBmVk3cNX/PscfZjUXsmwnAjOzbuD7d8/hwTnLClm2E4GZWck5EZiZlZwTgZlZyTkRmJmVnBOBmVnJORGYmZWcE4GZWck5EZiZlZwTgZlZyTkRmJmVnBOBmVnJORGYmZWcE4GZWck5EZiZlZwTgZlZyTkRmJmVnBOBmVk3EERhy3YiMDPrJlTQcp0IzMxKzonAzKzknAjMzErOicDMrOScCMzMSq7QRCDpDEkzJc2WdFk75Y6XtFXSu4uMx8zMdlZYIpDUG7gaOBMYD5wnaXwb5b4KTC0qFjMza1uRZwQnALMjYm5EbAYmA2fXKPdR4FfAkgJjMTOzNhSZCEYA83PDC9K47SSNAP4OmNTegiRdKGm6pOnNzc2dHqiZWZkVmQhq/Qiu+jfSVwGfjoit7S0oIq6JiAkRMaGpqanTAjQzM+hT4LIXAKNywyOBhVVlJgCTJQEMA86S1BIRvykwLjMzyykyEUwDxkkaC7wETATeky8QEWMr7yX9GLjFScDMrLEKSwQR0SLpErK7gXoD10XEDEkXpentXhcwM7PGKPKMgIi4Fbi1alzNBBAR7y8yFjMzq82/LDYzKzknAjOzknMiMDPrBqK4B5Q5EZiZdRcq6BFlTgRmZiXnRGBmVnJOBGZmJedEYGZWck4EZmYl50RgZlZyTgRmZiXnRGBmVnJOBGZmJedEYGZWck4EZmYl50RgZlZyTgRmZiXnRGBmVnJOBGZmJedEYGZWck4EZmbdQIEPKHMiMDPrLkQxjyhzIjAzKzknAjOzknMiMDMrOScCM7OScyIwMys5JwIzs5JzIjAzKzknAjOzknMiMDMrOScCM7OScyIwMys5JwIzs5IrNBFIOkPSTEmzJV1WY/rZkp6U9Lik6ZJOKjIeMzPbWZ+iFiypN3A1cDqwAJgmaUpEPJ0rdhcwJSJC0jHATcBRRcVkZmY7K/KM4ARgdkTMjYjNwGTg7HyBiFgbEZVutgdQbJfbZmZWQ5GJYAQwPze8II1rRdLfSXoW+B1wQa0FSbowNR1Nb25uLiRYM7OyKjIR1HqCwk7f+CPi1xFxFPAO4Iu1FhQR10TEhIiY0NTU1Mlhmpnt/XY0nnS+IhPBAmBUbngksLCtwhFxH3C4pGEFxmRm1m2pmAeUFZoIpgHjJI2VtC8wEZiSLyDplVK2apKOA/YFlhUYk5mZVSnsrqGIaJF0CTAV6A1cFxEzJF2Upk8C3gX8o6QtwAbg3Cjy/MfMzHZSWCIAiIhbgVurxk3Kvf8q8NUiYzAzs/b5l8VmZiXXYSJIt21eLGloIwIyM7PGqueMYCJwCNkvgydLelvlAq+ZmXV/HSaCiJgdEVcARwA3ANcBL0r6gqQDig7QzMyKVdc1gtQP0DeBrwO/At4NrAZ+X1xoZmbWCB3eNSTpUWAl8EPgsojYlCY9LOkvigzOzMyKV8/to+dExNxaEyLinZ0cj5mZNVg9TUMfkrR/ZUDSUElfKjAmMzNroHoSwZkRsbIyEBErgLOKC8nMzBqpnkTQW1LfyoCk/YC+7ZQ3M7NupJ5rBNcDd0n6EVk30hcA/11oVGZm1jAdJoKI+JqkPwOnkj1j4IsRMbXwyMzMrCHq6nQuIm4Dbis4FjMz6wL19DX0TkmzJK2StFrSGkmrGxGcmZlliuyfv54zgq8BfxMRzxQYh5mZdaCoTt7quWtosZOAmVnPVc8ZwXRJPwd+A1S6lyAibi4sKjMza5h6EsFgYD3w1ty4AJwIzMx6gHpuH/1AIwIxM7OuUc9dQ0dIukvSU2n4GEmfLT40MzNrhHouFv8XcDmwBSAiniR7apmZmfUA9SSC/hHxSNW4liKCMTOzxqsnESyVdDjp9wyS3g0sKjQqMzNrmHruGroYuAY4StJLwPPA+YVGZWZmDVPPXUNzgdMkDQB6RcSa4sMyM7NGqeeZxZ+vGgYgIv6toJjMzKyB6mkaWpd73w94O+AuJ8zMeoh6moa+mR+W9A1gSmERmZlZQ9Vz11C1/sBhnR2ImZl1jXquEfyZHV1h9waaAF8fMDPrIeq5RvD23PsWsm6p/YMyM7Meop5EUH276ODKnUMAEbG8UyMyM7OdRIGPKKsnETwGjAJWkD0gZ3/gxTQt8PUCM7PGUDHPKKvnYvHtZI+qHBYRB5I1Fd0cEWMjot0kIOkMSTMlzZZ0WY3p75X0ZHo9KOm1u7caZma2u+pJBMdHxK2VgYi4DXhzRzNJ6g1cDZwJjAfOkzS+qtjzwJsj4hjgi2RdWZiZWQPV2+ncZyWNkXSopCuAZXXMdwIwOyLmRsRmYDJwdr5ARDwYESvS4B+BkbsSvJmZ7bl6EsF5ZLeM/jq9mtK4jowA5ueGF6RxbfkgcFsdyzUzs05Uzy+LlwMflzQwItbuwrJrXdWoed1b0l+RJYKT2ph+IXAhwOjRo3chBDMz60g9j6p8k6SngafT8Gslfb+OZS8gu9uoYiSwsMbyjwGuBc6OiJpNThFxTURMiIgJTU1NdVRtZmb1qqdp6NvA20jXBSLiCeDkOuabBoyTNFbSvmSPt2zVR5Gk0cDNwD9ExHO7EriZmXWOen5HQETMV+v7V7fWMU+LpEuAqWRdU1wXETMkXZSmTwI+DxwIfD8tvyUiJuzaKpiZ2Z6oJxHMl/QmINI3+49RZzfU6bbTW6vGTcq9/xDwofrDNTOzzlZP09BFZI+rHEHW7n9sGjYzsx6g3TOC9KOwqyLivQ2Kx8zMGqzdM4KI2Ao0pSYhMzPrgeq5RjAPeEDSFHKPrYyIbxUVlJmZNU6bZwSSfprengvcksoOyr3MzKwHaO+M4PWSDiXrcvp7DYrHzMwarL1EMImsC+qxwPTceOHnEJiZ9RhtNg1FxHcj4lXAjyLisNyrw+cQmJlZ99Hh7wgi4p8bEYiZmbWvmOeT1feDMjMz68GcCMzMSs6JwMys5JwIzMxKzonAzKzknAjMzErOicDMrOScCMzMSs6JwMys5JwIzMxKzonAzKzknAjMzErOicDMrOScCMzMSs6JwMys5JwIzMxKzonAzGwvFxGFLt+JwMysm1BBjygrTSJ46qVV/Otvn2LJ6o1dHYqZ2V6lNInghWXr+e+HXmDF+i1dHYqZ2V6lNImgqFMqM7PurjSJoCIo9qKLmVl3U5pEUDkhKPjiu5lZt1OaRGBmZrWVJhFUrhH4jMDMrLVCE4GkMyTNlDRb0mU1ph8l6SFJmyR9qshYdjQOmZlZXp+iFiypN3A1cDqwAJgmaUpEPJ0rthz4GPCOouKo5ovFZmatFXlGcAIwOyLmRsRmYDJwdr5ARCyJiGlA4Tf3u2nIzKy2IhPBCGB+bnhBGrfLJF0oabqk6c3NzbsVjBuGzMxqKzIR1Dr27tb38Yi4JiImRMSEpqamPQzLzMzyikwEC4BRueGRwMIC62uXUtuQm4bMzForMhFMA8ZJGitpX2AiMKXA+trlpiEzs9oKu2soIlokXQJMBXoD10XEDEkXpemTJL0CmA4MBrZJuhQYHxGrC4vLdw2ZmbVSWCIAiIhbgVurxk3KvX+ZrMmocL5ryMysttL9stjMrLsp+gtsaRJBhU8IzKy7UkFXO0uTCCobsOhnf5qZdTelSQS+bcjMrLbyJILE5wNmZq2VJhH4wTRmZrWVJhGYmVltpUkE2n7/qE8JzMzyypMI0l83DZmZtVaaRGBmZrWVJhFs72Kia8MwM9vrlCcR4G6ozcxqKU0iMDOz2kqTCHb0PupTAjOzvPIkgvTXacDMrLXSJALr+Ta3bGP95pauDsOs2ylPIvCDaXq8d1z9AOM/P7Wrw6jb1m3BR2/8E0/MX9nVoVjJlSYRbL9ryI1DPdbTiwp7wmkhFq3awP88sZCP/Oyxrg7FSq40iaBM7nx6MUd//nY2bN7a1aHYXuot37yHq++e3dVhWJ2K/vpamkRQpq6Gvnb7s6zbvJX5K9Z3dSi2l5rbvI6vT53Z1WHYLirqkbvlSQTpbwnygJnZLilNIiiTSrLzhXFrpIjgn34ynbtnLunqUGwXlSYRVLqh9sHRrDh3Pr2YD/xoWleHYbuoNImgogx3DVWawYpqT7TO1VN+7d5DVqOUSpMIfFA0K9ae5oGNW7Yyb+m6TonFdk15EkH6628tZsXY0zObS254jFO+cQ9btm7rpIisXqVJBGXii8XWHd333FIAtnnHbbjSJAI/mMb2Vuoh7Zb+3+q+SpMI2P5gmp6/u/pisXWFRv9rPbd4DWd+5w+s3rilsRX3QCVKBOWxNzQN3T9rKS1u6y2Vzrojr9799lt3PMczi1bzwKylnVJvmZUmEewtTUOnfvMevnTL010cRbEenL2U83/4MN/7vfuyqUdPOUvtqtXoGVuva5UnEVTedPFeM6d5Hdfe/3yhdXR109CSNZsAmLesZ90K+NziNRxxxW0scB9OhershPLKz9zKO7//QOcutIcpTSIw21OTH5nP5q3buGPG4q4OpUfr7B99tmwLHnvRz3xoT2kSwfYuJhpwSnD8l/+Xr93+7C7PN23ecu59rrmAiKwzVPadvakp4qWVG/j5tBe7Ogyg+92uvKllK6/63O38zxMLuzqULldoIpB0hqSZkmZLuqzGdEn6bpr+pKTjCosl/S1iZ/3JQ/O49g9ztw83r9nE9++Zs8vLOWfSQ7zvukf2OJ4iLhav29TCmMt+x88efqHzFgr8Yvp8Zi1e06nLLJPzr32YT//qz6zZC+6cafTF4j3VvGYTG7Zs5Su3tf7SNm3ecm5+bEFjgthLFJYIJPUGrgbOBMYD50kaX1XsTGBcel0I/GdR8XSGLVu38cyi1Uy6dw7fumMmYy77Has3buHzv53Bl373TM3yk+6dw6aWth8QM2PhKpav21xz2iU3PMYnbnocgFXrtzTsebwvr9rIg7Nb34mxePVGAP7rvrm1ZmllV65N/J9fPsnp376v1bg1G7ewbVvXfL2ct3QdX/ifGbtV/6MvrGjzs+wsW6viWpquxxS5uR6cs5RTvn53pzzoaNX6LR0mrXpXpbOugVVfrD9n0kN84qYnWk2/6KeP8kD6n1i7qWX740U3t2xjc0v3vztORd2xIOlE4MqIeFsavhwgIv5frswPgHsi4sY0PBM4JSIWtbXcCRMmxPTp03c5nicXrORv/+MBDh7SjwF9++zy/AtWrGfjlp0/8OGD+7J49aYO5x91wH7069ObWUvWAjDuoIEArYbrmQZw8JB+DGxnHSplDxrUlyH77dNhbB3FAdnOv2jVxlbjl67dxMoNWzi8aSCqsazqZXRUL8CGLVtZsGJDXfN2tKx6vLxqI2s2tTB22AD69NL2ZfQSHN7UejntrVcAs3ex/k0t23hx+fq655m7dB1btwUHDNiXAwfs2yqmw5oG0DsdGSvjXtnOMncl1sryhvbfh2ED+7ZZblsEc5rXtbvc9j6jWuuSn9a3Ty9GH9B/p/K1Pqt66tzYspX5y3fe16rLV5erTB99QP/tn9+hB/bnhWXr6aUsKb9icD8G9dv140xbKvvXJ04/go+dOm63liHp0YiYUGta50W6sxHA/NzwAuANdZQZAbRKBJIuJDtjYPTo0bsVzBHDBzHx+FGs2bh736pHH9Cf3z+7o5/1Q4b0Y+GqjRw3eih3PbuElq3b2BbwmhFD+PNLqxg5dD+OHD6Iu9I8Y4cNZGDfLBEMH9yXccN3HOxHHbAf44YPZOHKDazbvHX7tDnNa+nTuxfjhmc73+B+fVi9sYVRQ/szbNC+bcbav28fnpi/ktcfOrTub02zlqxln95i3PCBbNm6jXnL1m+Po2LRn1/mtSOHMGLoftvHrVi/hSOqyh3WNICpMxbzpsMPZP/+7SeiWUvWMubA/tvrioAFKzbw6hGDW/3T12Plhi00r9m0U9ztqWyrI4cPolcvGNC3D4/PX8nbjn7FTtvu4P33477nmjntVQexb5+dT6ZnL1nLYU0Ddqn+F5ev5zUjhjDqgP06LNs0qC8PzlnGcaOHsm+fLLg+vXvxzKLVHPWKQdvLzVqylqZBfTly+KC2FsXsJWsZOXS/umId2K8Pf3pxJW887MAO96c5zes4YvjANpPQrCVr2W+f3jXr7d1LPPvymlbrkp/vkP1bxztsYF8emruM08cPp3evtgObtWQt+/ffp2ad85dv2Gn7r1i/haVrW+9H85dv4Mjhgzj8oAHb43z1iMEsWrWBlm3B+IMH88Ky9Rx9yI7//4MGt500d8dRrxjEW48e3qnLrCgyEdT6ZKpPP+opQ0RcA1wD2RnB7gTTb5/efOVdx+zOrGY9ztVdHYDtVYq8WLwAGJUbHglUX56vp4yZmRWoyEQwDRgnaaykfYGJwJSqMlOAf0x3D70RWNXe9QEzM+t8hTUNRUSLpEuAqUBv4LqImCHpojR9EnArcBYwG1gPfKCoeMzMrLYirxEQEbeSHezz4ybl3gdwcZExmJlZ+0rzy2IzM6vNicDMrOScCMzMSs6JwMys5ArrYqIokpqB3e35bBjQVY8z6qq6vc7lqNvr3PPr3dO6D42IploTul0i2BOSprfV10ZPrdvrXI66vc49v94i63bTkJlZyTkRmJmVXNkSwTUlrNvrXI66vc49v97C6i7VNQIzM9tZ2c4IzMysihOBmVnJlSYRSDpD0kxJsyVdVnBd10laIump3LgDJN0paVb6O7SAekdJulvSM5JmSPp4I+qW1E/SI5KeSPV+oRH15urvLelPkm5pcL3zJP1Z0uOSpjeqbkn7S/qlpGfTZ31ig+o9Mq1r5bVa0qUNqvtf0r71lKQb0z7XqM/546neGZIuTeM6ve5dPW5Iujwdz2ZKetue1F2KRCCpN9lDmc4ExgPnSRpfYJU/Bs6oGncZcFdEjAPuSsOdrQX4ZES8CngjcHFaz6Lr3gS8JSJeCxwLnJGeL9GIdQb4OPBMbrhR9QL8VUQcm7u3uxF1fwe4PSKOAl5Ltu6F1xsRM9O6Hgu8nqzr+F8XXbekEcDHgAkR8Wqybu0nFl1vqvvVwD8BJ5Bt67dLGldQ3T+mzuNG+r+eCByd5vl+Os7tnojo8S/gRGBqbvhy4PKC6xwDPJUbngkcnN4fDMxswHr/Fji9kXUD/YHHyJ5PXXi9ZE+1uwt4C3BLI7c1MA8YVjWu0LqBwcDzpBs9umr/At4KPNCgda482/wAsq7zb0n1N2L/Oge4Njf8OeD/FlV3vceN6mMY2XNfTtzdektxRsCOHaliQRrXSMMjPX0t/T2oyMokjQFeBzzciLpT88zjwBLgzohoSL3AVWT/mNty4xq1rQO4Q9Kjki5sUN2HAc3Aj1Jz2LWSBjSg3moTgRvT+0LrjoiXgG8ALwKLyJ5keEfR9SZPASdLOlBSf7IHaY1qUN20U0+nHtPKkghUY1yPvW9W0kDgV8ClEbG6EXVGxNbImgxGAiekU+pCSXo7sCQiHi26rjb8RUQcR9bkeLGkkxtQZx/gOOA/I+J1wDqKbfraibJHz/4t8IsG1TcUOBsYCxwCDJB0fiPqjohngK8CdwK3A0+QNcF2tU49ppUlESwgy+IVI4GFDY5hsaSDAdLfJUVUImkfsiTws4i4uZF1A0TESuAesnbLouv9C+BvJc0DJgNvkXR9A+oFICIWpr9LyNrKT2hA3QuABemMC+CXZImhYZ8xWeJ7LCIWp+Gi6z4NeD4imiNiC3Az8KYG1AtARPwwIo6LiJOB5cCsRtXdTj2dekwrSyKYBoyTNDZ9m5kITGlwDFOA96X37yNrv+9UkgT8EHgmIr7VqLolNUnaP73fj+wf99mi642IyyNiZESMIftMfx8R5xddL4CkAZIGVd6TtVk/VXTdEfEyMF/SkWnUqcDTRddb5Tx2NAvRgLpfBN4oqX/ax08lu0DekHWWdFD6Oxp4J9m6N2p7t1XPFGCipL6SxgLjgEd2u5bOvriyt77I2vaeA+YAVxRc141kbZlbyDL3B4EDyS5qzrS+2gcAAAN/SURBVEp/Dyig3pPITg+fBB5Pr7OKrhs4BvhTqvcp4PNpfOHrnIvhFHZcLG7Etj6MrJngCWBGZZ9qUN3HAtPT9v4NMLRR25rsZoBlwJDcuEas8xfIvlw8BfwU6NvAdf4DWbJ9Aji1qHXe1eMGcEU6ns0EztyTut3FhJlZyZWlacjMzNrgRGBmVnJOBGZmJedEYGZWck4EZmYl50Rg1gCpl87+XR2HWS2+fdSsDZL6RESndCeQfv08ISKWdsbyzDqTzwisR5M0JvXb/9+Snkz9+PeX9HpJ96YO46bmfsZ/j6R/l3Qv8HFJx0t6UNmzFh6RNCh1sPd1SdPSMj+c5j0lzV95VsDPlPkYWR85d0u6O5X9T0nTlXt+Qxp/Vpr3fknf1Y5nLAxQ1l/9tNTZ3NkN35jWY/Xp6gDMGuBI4IMR8YCk64CLgb8Dzo6IZknnAl8GLkjl94+IN6fuSJ4Fzo2IaZIGAxvIfvG5KiKOl9QXeEDSHWne15H1Eb8QeICsY7rvSvoE2bMLKmcEV0TE8tSH/F2SjiH75fsPgJMj4nlJ+W4criDrQuOC1J3HI5L+NyLWFbC9rGScCKwM5kfEA+n99cBngFcDd2Zd19Cb7Kf9FT9Pf48EFkXENIBIPblKeitwjKR3p3JDyPp62Qw8EhELUrnHyfqXv79GTH+fuq7uQ9bP/HiyM/S5EfF8KnMjUOne+q1kHex9Kg33A0bT+oE8ZrvFicDKoPpC2BpgRkSc2Eb5yrds1Zi3Mv6jETG11UjpFLKntVVspcb/WOok7FPA8RGxQtKPyQ7stboWztf5roiY2U4Zs93iawRWBqMlVQ765wF/BJoq4yTtI+noGvM9Cxwi6fhUbpCkPmRPg/rn1OU3ko5IPZC2Zw0wKL0fTJZsVkkaTtatc6W+w5Q9VAjg3Nz8U4GPpt43kfS6jlfbrD4+I7AyeAZ4n6QfkPXi+D2yA+t3JQ0h+z+4iqwX0e0iYnO6fvC91L32BrIutq8la/J5LB2Ym4F3dBDDNcBtkhZFxF9J+lOqby7ZtQQiYoOkjwC3S1pK626Fv5hifDLVOQ94++5sDLNqvn3UerT07fqWyB56vteTNDAi1qaD/dXArIj4dlfHZT2bm4bM9i7/lC4yzyC7CP2DLo7HSsBnBGZmJeczAjOzknMiMDMrOScCM7OScyIwMys5JwIzs5L7/ws7zHkQIPofAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAEWCAYAAABrDZDcAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deZwdVZ338c+XDknIRgJptiwkYARxHjYD6oiCArLIM1HHBdxmRh0GBwRneWaCzvjMDI6OozIKg0ZExMcF3FDz0rANCqMwSgIGJIRASCLpgNlJyJ5Ofs8fdW5SfVPdfdPpup3u+r5fr371rapTdX51l/rdOqfuKUUEZmZWXQf0dQBmZta3nAjMzCrOicDMrOKcCMzMKs6JwMys4pwIzMwqzonAOiXpo5Ju6oN6l0g6p9nr1m3nLEltuel5ks7a1+3uC0mTJIWkQZ0sD0kv6aW67pP0wd7Y1j7E0Gv7Y11zIrBORcQnI6JPDwb7i4h4eUTcty/bkPRPkr7ZSyE1laQjJX1F0nOSNkhaJOkWScf3dWy275wIzKxLkg4FHgSGAa8FRgKnAvcD53ayTuFZi+2fnAgqRtLfS1om6UVJCySd3UXZDt9gJZ0h6UFJL0haKulP0/xbJH1R0h3p2+IDko6Q9HlJayU9KemULup5r6TfSVot6WN1yw6QNF3SM2n5dyUd0uC6t0j6RG66vrlniaSrJT2R4vyapKGdxLiryUlSS2o2eyY9jw9LmpCWfSE9N+vT/Nem+ecDHwXemZ6jR9P8gyV9VdLz6XX5hKSWXD2flbRK0iLgTZ09hzkXpm/rqyR9RtKuz7ik90uan/b1LklH55adm16ndZL+E1Bum38FrAfeGxHPROaFiPhaRFyf1q81W31A0rPAz9L870n6fdruf0t6ed3rM0PSPel5vD8fU3KOpKdTzDdIEtbrnAgqRNJxwBXAaRExEjgPWNLguhOBO4DrgVbgZGBursg7gH8AxgJbgf8BHknT3weu7WS7JwBfAt4LHAUcCozPFbkSeDNwZlq+FrihwXUb8W6y5+FY4KVpH7rz18AlwIXAKOD9wKa0bDbZc3MI8G3ge5KGRsSdwCeB70TEiIg4KZX/OtAOvAQ4BXgjUGuO+3PgojR/KvC2BmJ7Syp7KjAtxYakN5MloreSvX6/AG5Ny8YCP2D36/cM8JrcNs8BfhgROxuo/0zgZWTPKWTvmSnAYWTvh2/VlX83cE2qd27B8ouA04CTyN5j52G9LyL8V5E/soPNCrIP9oENlP8n4Jvp8dVkB4OicrcAX8lNfxiYn5v+X8ALnaz7ceC23PRwYBtwTpqeD5ydW34ksB0Y1MC6twCfyC0/C2jLTS8BLstNXwg800XZ2nYXANMafM7XAifVP59p+nCypHlQbt4lwM/T45/VxfdGIIBBndQVwPm56b8E7k2P7wA+kFt2AFnyOhp4H/Cr3DIBbcAH0/TCujj+CHgBeBG4O82blOo/povnYnQqc3Du9cm/fiOAHcCE3P6ckVv+XWB6X3+OBuKfzwgqJCIWAh8hOyCtkHSbpKMaXH0C2TfFzizPPd5cMD2ik/WOApbmYtwIrM4tPxr4YWqOeoEsMewgO4h2t24jluYe/y5tszudPheS/iY1v6xL8R5M9m23yNHAgcDzuf37Mtm3Z6jbvxRfdzrbn6OBL+TqWUN2wB9XX09kR938dlaTJeDa8pkRMZqsyWhwZ/Wnpq1/S01o69l99jm2qHxEbEhx5V+D3+ceb6Lz95HtAyeCiomIb0fEGWQHhgA+3eCqS8maT3rb82QHVgAkDSNr4snXe0FEjM79DY2IZQ2su5Gsg7PmiIL6J+QeTwSeayDmwuci9Qf8PVkTxph0sFzH7vb2+qF+l5KdEYzN7duoiKi1o3fYvxRfdzrbn6XAX9Q9jwdFxIP19aR2+Px27gXenO9v6EJ+H99F1jx1DllCnFSroiheSSPImtQaeQ2sFzkRVIik4yS9QdIQYAvZN/UdDa7+LbKOu3dIGiTpUEkn90JY3wcuUtYRPRj4Fzq+L2cA/1rrRJTUKmlag+vOJes8PUTSEWRnQ/UulzQ+dUB/FPhOAzHfBFwjaYoyJyq7smYkWXv/SmCQpI+T9SHULAcm1Q6oEfE8cDfwOUmjlHWMHyvpzFT+u8CVKb4xwPQGYvs/ksakzuurcvszA7i61lmbOqnfnpb9FHi5pLcqu9rnSjomzWuBMcA3UnySNJKsL6QrI8kS3WqyhPzJgjIX5l6/a4BfR8TSgnJWIieCahkC/BuwiuyU+zCyg1+3IuJZsjb0vyE7fZ9L1oG315RdXfTRtN15wOVkHavPk7Wpt+WKfwGYCdwt6UXgV8ArG1z3G8CjZE0Sd1N8kP92WrYo/X2ioEy9a8kO0neTXU3zVeAg4C6ytvinyJplttCxieV76f9qSY+kx+8ja155IsX/fXY3w3wlbfNRso7W2/NBpCtuZtTF9mPgYbLX56cpNiLih2Rnf7elZprHgQvSslXA28neG6vJOncfqG0wLX9V2p9fkvUNzCU70H+oi+fp/6XnYVnav18VlPk28H/J3lOvIOs8tiZT6oQxqxxJS8g6RP+rr2OpIkm3kHXIN3KllpXIZwRmZhXnRFBx2v0jsPq/hpqMzKz/c9OQmVnF+YzAzKzi+t3AUGPHjo1Jkyb1dRhmZv3Kww8/vCoiWouW9btEMGnSJObMmdPXYZiZ9SuSOv1lupuGzMwqzonAzKzinAjMzCrOicDMrOKcCMzMKs6JwMys4pwIzMwqrjKJ4KnlL3Lt3QtYtWFrX4diZrZfqUwieHr5Bq772ULWbNzW16GYme1XKpMIzMysmBOBmVnFORGYmVVc5RKBb79gZtZRqYlA0vmSFkhaKGl6J2XOkjRX0jxJ95cZj5mZ7am0YagltQA3AOcCbcBsSTMj4olcmdHAF4HzI+JZSYeVF09ZWzYz69/KPCM4HVgYEYsiYhtwGzCtrsy7gNsj4lmAiFhRYjxmZlagzEQwDliam25L8/JeCoyRdJ+khyW9r2hDki6VNEfSnJUrV5YUrplZNZWZCIoaY+q7agcBrwDeBJwH/KOkl+6xUsSNETE1Iqa2thbeac3MzHqozFtVtgETctPjgecKyqyKiI3ARkn/DZwEPFVWULFHLjIzq7YyzwhmA1MkTZY0GLgYmFlX5sfAayUNkjQMeCUwv4xg3FdsZlastDOCiGiXdAVwF9AC3BwR8yRdlpbPiIj5ku4EHgN2AjdFxONlxWRmZnsqs2mIiJgFzKqbN6Nu+jPAZ8qMw8zMOle5XxabmVlHlUsEHmLCzKyjyiUCMzPrqDKJwENMmJkVq0wiMDOzYk4EZmYV50RgZlZxlUsEvmrIzKyjCiUC9xabmRWpUCIwM7MiTgRmZhXnRGBmVnGVSwS+H4GZWUeVSwRmZtZRZRKBh5gwMytWmURgZmbFnAjMzCrOicDMrOIqlwg8xISZWUeVSQTuKzYzK1aZRGBmZsWcCMzMKq7URCDpfEkLJC2UNL1g+VmS1kmam/4+XmY8Zma2p0FlbVhSC3ADcC7QBsyWNDMinqgr+ouIuKisOMzMrGtlnhGcDiyMiEURsQ24DZhWYn1mZtYDZSaCccDS3HRbmlfv1ZIelXSHpJcXbUjSpZLmSJqzcuXKHgUjjzFhZlaozERQdOStv4r/EeDoiDgJuB74UdGGIuLGiJgaEVNbW1t7OUwzs2orMxG0ARNy0+OB5/IFImJ9RGxIj2cBB0oaW2JMZmZWp8xEMBuYImmypMHAxcDMfAFJRyi12Ug6PcWzusSYzMysTmlXDUVEu6QrgLuAFuDmiJgn6bK0fAbwNuBDktqBzcDFEeUOAuEhJszMOiotEcCu5p5ZdfNm5B7/J/CfZcZQ465iM7Ni/mWxmVnFORGYmVWcE4GZWcVVLhHEHj9lMDOrtsolAjMz66gyicAjTJiZFatMIjAzs2JOBGZmFedEYGZWcZVLBB5iwsyso8okAncWm5kVq0wiMDOzYk4EZmYV50RgZlZxlUsE7is2M+uoconAzMw6qkwikG9NY2ZWqDKJwMzMijkRmJlVnBOBmVnFVS4RhMeYMDProDqJwH3FZmaFqpMIzMysUKmJQNL5khZIWihpehflTpO0Q9LbyozHzMz2VFoikNQC3ABcAJwAXCLphE7KfRq4q6xYzMysc2WeEZwOLIyIRRGxDbgNmFZQ7sPAD4AVJcayi7uKzcw6KjMRjAOW5qbb0rxdJI0D3gLM6GpDki6VNEfSnJUrV/Z6oGZmVVZmIii6Tqf+C/nngb+PiB1dbSgiboyIqRExtbW1tdeCMTMzGFTittuACbnp8cBzdWWmArcpu33YWOBCSe0R8aMS4zIzs5wyE8FsYIqkycAy4GLgXfkCETG59ljSLcBPnATMzJqrtEQQEe2SriC7GqgFuDki5km6LC3vsl/AzMyao8wzAiJiFjCrbl5hAoiIPy0zlt31NKMWM7P+o9vO4nS1zuWSxjQjoLKkfggzM6vTyFVDFwNHAbMl3SbpPPmoamY2YHSbCCJiYUR8DHgp8G3gZuBZSf8s6ZCyAzQzs3I19DsCSScCnwM+Q/Yr4LcB64GflReamZk1Q7edxZIeBl4AvgpMj4itadGvJb2mzODK4d5iM7O8Rq4aentELCpaEBFv7eV4zMysyRppGvqgpNG1CUljJH2ixJhK4d5tM7NijSSCCyLihdpERKwFLiwvJDMza6ZGEkGLpCG1CUkHAUO6KG9mZv1II30E3wTulfQ1sp7W9wNfLzUqMzNrmm4TQUT8u6TfAmeTNbVfExH99m5iHmLCzKyjhsYaiog7gDtKjqVU/i20mVmxRsYaequkpyWtk7Re0ouS1jcjODMzK18jZwT/DvzviJhfdjBmZtZ8jVw1tNxJwMxs4GrkjGCOpO8APwJqw0sQEbeXFlWJ3FdsZtZRI4lgFLAJeGNuXgD9MhGYmVlHjVw++mfNCKRs8iATZmaFGrlq6KWS7pX0eJo+UdI/lB+amZk1QyOdxV8Brga2A0TEY2R3LTMzswGgkUQwLCIeqpvXXkYwZmbWfI0kglWSjiVdcCPpbcDzpUZVIg8xYWbWUSNXDV0O3AgcL2kZsBh4T6lRlcBDTJiZFWvk5vWLIuIcoBU4PiLOiIgljWxc0vmSFkhaKGl6wfJpkh6TNFfSHEln7PUemJnZPmnknsUfr5sGICL+pZv1WoAbgHOBNmC2pJkR8USu2L3AzIgISScC3wWO36s9MDOzfdJIH8HG3N8O4AJgUgPrnQ4sTGcU24DbgGn5AhGxIWJXq/1w/MNfM7Oma+QHZZ/LT0v6LDCzgW2PA5bmptuAV9YXkvQW4FPAYcCbijYk6VLgUoCJEyc2UHXnwr3FZmYdNHJGUG8YcEwD5Yq6Z/c4CkfEDyPieODNwDVFG4qIGyNiakRMbW1t3atgzcysa430EfyW3QfwFrJO4y77B5I2YEJuejzwXGeFI+K/JR0raWxErGpg+3vFFw2ZmRVr5PLRi3KP28mGpW7kB2WzgSmSJgPLyH6N/K58AUkvAZ5JncWnAoOB1Q1FbmZmvaKRRPBi3fQo5S7Kj4g1RStFRLukK4C7yM4kbo6IeZIuS8tnAH8MvE/SdmAz8M5wI76ZWVM1kggeIWviWUvWwjIaeDYtC7roL4iIWcCsunkzco8/DXx670I2M7Pe1Ehn8Z1kt6ocGxGHkjUV3R4RkyOikU7j/YpPN8zMOmokEZyWvtkDEBF3AGeWF1JJ3FtsZlaokaahVen+A98k+0L9Htyha2Y2YDRyRnAJ2SWjP0x/rWmemZkNAI38sngNcJWkERGxoQkxmZlZEzVyq8o/lPQE8ESaPknSF0uPrCS+ONXMrKNGmob+AziP1C8QEY8CryszKDMza56GxhqKiKV1s3aUEEup5MuGzMwKNXLV0FJJfwiEpMHAlcD8csMyM7NmaeSM4DKy21WOIxtI7uQ0bWZmA0CXZwTpLmOfj4h3NykeMzNrsi7PCCJiB9CamoQGhPAgE2ZmHTTSR7AEeEDSTLLbVQIQEdeWFVQZ5L5iM7NCnZ4RSPpGevhO4Cep7Mjcn5mZDQBdnRG8QtLRZENOX9+keMzMrMm6SgQzyIagngzMyc0X3dyHwMzM+o9Om4Yi4rqIeBnwtYg4JvfXL+9DsIv7is3MOuj2dwQR8aFmBGJmZn2joSEmBgJfNGRmVqwyicDMzIo5EZiZVZwTgZlZxZWaCCSdL2mBpIWSphcsf7ekx9Lfg5JOKjMe8EVDZmb1SksEacC6G4ALgBOASySdUFdsMXBmRJwIXAPcWGI8ZW3azKxfK/OM4HRgYUQsiohtwG3AtHyBiHgwItamyV8B40uMx8zMCpSZCMYB+TubtaV5nfkAcEeJ8ZiZWYFGRh/tqaK2mMImekmvJ0sEZ3Sy/FLgUoCJEyf2VnxmZka5ZwRtwITc9HjgufpCkk4EbgKmRcTqog1FxI0RMTUipra2tu5TUOHeYjOzDspMBLOBKZImpxvbXAzMzBeQNBG4HXhvRDxVYixmZtaJ0pqGIqJd0hXAXUALcHNEzJN0WVo+A/g4cCjwxXRVT3tETC0jHl80ZGZWrMw+AiJiFjCrbt6M3OMPAh8sMwYzM+uaf1lsZlZxTgRmZhVXuUQQHmTCzKyDyiQC9xWbmRWrTCIwM7NiTgRmZhXnRGBmVnGVSwQeYsLMrKPKJQIzM+uoMonAQ0yYmRWrTCIwM7NiTgRmZhXnRGBmVnGVSwS+aMjMrKMKJQL3FpuZFalQIjAzsyJOBGZmFedEYGZWcZVLBOExJszMOqhcIjAzs44qkwg8xISZWbHKJAIzMyvmRGBmVnGlJgJJ50taIGmhpOkFy4+X9D+Stkr62zJjMTOzYoPK2rCkFuAG4FygDZgtaWZEPJErtga4EnhzWXHU8zVDZmYdlXlGcDqwMCIWRcQ24DZgWr5ARKyIiNnA9hLjADzAhJlZZ8pMBOOApbnptjRvr0m6VNIcSXNWrlzZK8GZmVmmzERQ9CW8Ry0zEXFjREyNiKmtra37GJaZmeWVmQjagAm56fHAcyXWZ2ZmPVBmIpgNTJE0WdJg4GJgZon1Nca9xWZmHZR21VBEtEu6ArgLaAFujoh5ki5Ly2dIOgKYA4wCdkr6CHBCRKwvKy4zM+uotEQAEBGzgFl182bkHv+erMmodPIYE2a7XH/v03zunqdY/KkL/dkw/7LYrIo+d89TAHgwXgMnArNKqp0EOA8YOBGYVVKtMWinTwmMCiaC8Hcgs12cBwwqlAjcHWa2W62D2F+MDCqUCMy688KmbWzetqOvw2iK2hcjnxEYOBGY7XLyv9zDOdfe39dhNJUTQf9x8y8X89DiNaVs24nALGfZC5v7OoSmqF015M7i/uPf7niSnz25opRtVy4R+H1vBqLWR2D9RRCl3Xu9conAzHYLfzPqNyLKu+ilMonAv6I3y9nVNNS3YVjjgvKOY5VJBGa2267jiRNBvxERu5r0epsTgfWZtRu3sW5z6XcptS74dwT9R5lnBKWOPmrWlVOuuQcJFn/qTX0dSuXITUP9jvsIepH7xvYvfj36ljuL+5mSTgkqkwjKalsz649qnwefEfQPtYTtMwIz6zW7h6F2JugPaiduvmrIzHrNQLxqaOfO4Mdzl9G+Y2dfh9Lrai+TrxqyptjavoMPfn02Ty1/sa9DsRLVRh8dSE1DP5q7jKtum8tNv1zc16H0ul1NQz4j6B0D6H1fisfa1vFf81dw9e2/7etQrAkGUtPQ2k3ZpcjL12/p40h63+4zgnJULhFY1w7Y9U1x4Bwg9hdb23ewYz/7Cj6QXuYDapfE7mfPcW/yGcE+qsoQExu3tvO71Rt7vH5L+jQN5A9TXznuH+7kylt/09dhAAPzVpUHDMDmrprdncXuI7AG/NktsznzM/f1eP2W9EbbMYAOEPuTn/72+b4OIVO7amgAvcy7zghK2qkHn1nVZzcuKrsJr9REIOl8SQskLZQ0vWC5JF2Xlj8m6dQy4+nP7p2/nPVbuh+OoXbjip5eObHrF6cD78ILK9BXieCvvzOX13/2vl7dpkps1mxbu4l3feXX/N0PHuv1bTei314+KqkFuAG4ADgBuETSCXXFLgCmpL9LgS+VFU9fqf/l5uZtO9i4tX2PcrOXrOH+p1YWbmPF+i184Otz+PC3G29W2Ly9Z99cetJH8OKW7X3yC9Xr7n2aSdN/ul/8OnbJqo1s6eFzXu+mXyzi8WXrOsyLiF7dz123qizhm+bv122hbe2mLsvc/ptlLF7VdRPm3u5v7b3bk36Y7d18cdrani2vf106c88Ty/n+w217zN+xM9ja3vP3SVmXj6qsD5GkVwP/FBHnpemrASLiU7kyXwbui4hb0/QC4KyI6PT8eerUqTFnzpy9jufxZeu46PpfcuTBQxkxpJwhlnZGsHbTdkYPO5AWidUbt7F9x06OGDV0V5mnV2wAYMphIzqs29l8gPadsetDU7S8aDvHjB2+q70/b/n6LYwZPpjBLcXfAba27+TZNZsaqqsn5Yti7W69jVvb2RkwcmjH1622/rGtwzsksKVrN3P4qCEMHdRSSjz1dkSwaGX3r08j298ZwTMF23p+3RaGDDqAQ4YP3qvYuotl0qHDOLCT90IjVry4lTHDDuywjUb2s7sym7btYOWGrYwfc9Cu5sru/G71JralA/revIa1z9fQAw9gwphhhWW279jJktWNv8978jnvSu19Mf2C47nszGP3at0aSQ9HxNSiZWUOOjcOWJqbbgNe2UCZcUCHRCDpUrIzBiZOnNijYF5y2AguOX1C6aNdzntuPZMPHc6QAw/gkA3b+PXiNbx2yu4XfeWGrby4pZ0ph3d8I6zdtJ21m7btMb9m8aqNnDJxNEcePLRwec1Bg1t4rG0dxx85snD51vadjB42mHGjO9/Os2s2cfrkQxg7orGDzrNrNjUUW73n122h5QB1us81W7bv5JmVG/YoN/TAFn67bB3HHdFxXxev2kjriCEcsZfxLFm9kUOGD+42niKLVm7kuMNHcuxhwzst8/SKDRzTOrzb7T+zciMvOWxEh3KDBx3A/OfX88pjDtnr2IqMGDqI3zz7AiccNWqfttO+Mxg3+iBGHbT7ULJm4zZe2Ly9y/3c0r6DpWs2d1pm+47gnieWc8zYEQwe1FgieMlhI7jj8d/z+uNaOWjw3n0JWLxqI8cdMarLz8WS1Zt4xdFjOHzUkG63t37Ldla8uLXwc75qw57zG3HCUQdz9vGH7fV6jSgzERS9evWnH42UISJuBG6E7IygJ8EMPbCFT731xJ6samY2oJXZWdwGTMhNjwee60EZMzMrUZmJYDYwRdJkSYOBi4GZdWVmAu9LVw+9CljXVf+AmZn1vtKahiKiXdIVwF1AC3BzRMyTdFlaPgOYBVwILAQ2AX9WVjxmZlas1DuURcQssoN9ft6M3OMALi8zBjMz65p/WWxmVnFOBGZmFedEYGZWcU4EZmYVV9oQE2WRtBL4XQ9XHwus6sVw+kPd3udq1O19Hvj17mvdR0dEa9GCfpcI9oWkOZ2NtTFQ6/Y+V6Nu7/PAr7fMut00ZGZWcU4EZmYVV7VEcGMF6/Y+V6Nu7/PAr7e0uivVR2BmZnuq2hmBmZnVcSIwM6u4yiQCSedLWiBpoaTpJdd1s6QVkh7PzTtE0j2Snk7/x5RQ7wRJP5c0X9I8SVc1o25JQyU9JOnRVO8/N6PeXP0tkn4j6SdNrneJpN9KmitpTrPqljRa0vclPZle61c3qd7j0r7W/tZL+kiT6v6r9N56XNKt6T3XrNf5qlTvPEkfSfN6ve69PW5IujodzxZIOm9f6q5EIpDUAtwAXACcAFwi6YQSq7wFOL9u3nTg3oiYAtybpntbO/A3EfEy4FXA5Wk/y657K/CGiDgJOBk4P91fohn7DHAVMD833ax6AV4fESfnru1uRt1fAO6MiOOBk8j2vfR6I2JB2teTgVeQDR3/w7LrljQOuBKYGhF/QDas/cVl15vq/gPgz4HTyZ7riyRNKanuW2jwuJE+1xcDL0/rfDEd53omIgb8H/Bq4K7c9NXA1SXXOQl4PDe9ADgyPT4SWNCE/f4xcG4z6waGAY+Q3Z+69HrJ7mp3L/AG4CfNfK6BJcDYunml1g2MAhaTLvToq/cX8EbggSbtc+3e5oeQDZ3/k1R/M95fbwduyk3/I/B3ZdXd6HGj/hhGdt+XV/e03kqcEbD7jVTTluY10+GR7r6W/pdzF+pE0iTgFODXzag7Nc/MBVYA90REU+oFPk/2wdyZm9es5zqAuyU9LOnSJtV9DLAS+FpqDrtJ0vAm1FvvYuDW9LjUuiNiGfBZ4FngebI7Gd5ddr3J48DrJB0qaRjZjbQmNKluuqinV49pVUkEKpg3YK+blTQC+AHwkYhY34w6I2JHZE0G44HT0yl1qSRdBKyIiIfLrqsTr4mIU8maHC+X9Lom1DkIOBX4UkScAmyk3KavPSi79ewfAd9rUn1jgGnAZOAoYLik9zSj7oiYD3wauAe4E3iUrAm2r/XqMa0qiaCNLIvXjAeea3IMyyUdCZD+ryijEkkHkiWBb0XE7c2sGyAiXgDuI2u3LLve1wB/JGkJcBvwBknfbEK9AETEc+n/CrK28tObUHcb0JbOuAC+T5YYmvYakyW+RyJieZouu+5zgMURsTIitgO3A3/YhHoBiIivRsSpEfE6YA3wdLPq7qKeXj2mVSURzAamSJqcvs1cDMxscgwzgT9Jj/+ErP2+V0kS8FVgfkRc26y6JbVKGp0eH0T2wX2y7Hoj4uqIGB8Rk8he059FxHvKrhdA0nBJI2uPydqsHy+77oj4PbBU0nFp1tnAE2XXW+cSdjcL0YS6nwVeJWlYeo+fTdZB3pR9lnRY+j8ReCvZvjfr+e6snpnAxZKGSJoMTAEe6nEtvd25sr/+kbXtPQU8A3ys5LpuJWvL3E6WuT8AHErWqfl0+n9ICfWeQXZ6+BgwN/1dWHbdwInAb1K9jwMfT/NL3+dcDGexu7O4Gc/1MWTNBI8C82rvqSbVfTIwJz3fPwLGNOu5JrsYYDVwcG5eM/b5n8m+XDwOfAMY0sR9/v5TLxkAAAMlSURBVAVZsn0UOLusfd7b4wbwsXQ8WwBcsC91e4gJM7OKq0rTkJmZdcKJwMys4pwIzMwqzonAzKzinAjMzCrOicCsCdIoncP6Og6zIr581KwTkgZFRK8MJ5B+/Tw1Ilb1xvbMepPPCGxAkzQpjdv/dUmPpXH8h0l6haT704Bxd+V+xn+fpE9Kuh+4StJpkh5Udq+FhySNTAPsfUbS7LTNv0jrnpXWr90r4FvKXEk2Rs7PJf08lf2SpDnK3b8hzb8wrftLSddp9z0Whisbr352GmxuWtOfTBuwBvV1AGZNcBzwgYh4QNLNwOXAW4BpEbFS0juBfwXen8qPjogz03AkTwLvjIjZkkYBm8l+8bkuIk6TNAR4QNLdad1TyMaIfw54gGxguusk/TXZvQtqZwQfi4g1aQz5eyWdSPbL9y8Dr4uIxZLywzh8jGwIjfen4TwekvRfEbGxhOfLKsaJwKpgaUQ8kB5/E/go8AfAPdnQNbSQ/bS/5jvp/3HA8xExGyDSSK6S3gicKOltqdzBZGO9bAMeioi2VG4u2fjyvyyI6R1p6OpBZOPMn0B2hr4oIhanMrcCteGt30g2wN7fpumhwEQ63pDHrEecCKwK6jvCXgTmRcSrOylf+5atgnVr8z8cEXd1mCmdRXa3tpodFHzG0iBhfwucFhFrJd1CdmAvGlo4X+cfR8SCLsqY9Yj7CKwKJkqqHfQvAX4FtNbmSTpQ0ssL1nsSOErSaancSEmDyO4G9aE05DeSXppGIO3Ki8DI9HgUWbJZJ+lwsmGda/Udo+ymQgDvzK1/F/DhNPomkk7pfrfNGuMzAquC+cCfSPoy2SiO15MdWK+TdDDZ5+DzZKOI7hIR21L/wfVpeO3NZENs30TW5PNIOjCvBN7cTQw3AndIej4iXi/pN6m+RWR9CUTEZkl/CdwpaRUdhxW+JsX4WKpzCXBRT54Ms3q+fNQGtPTt+ieR3fR8vydpRERsSAf7G4CnI+I/+jouG9jcNGS2f/nz1Mk8j6wT+st9HI9VgM8IzMwqzmcEZmYV50RgZlZxTgRmZhXnRGBmVnFOBGZmFff/AUJ76zLVL4AvAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "files = [\"s_8_cell.deduplicated.bedGraph\", \"s_epiblast.deduplicated.bedGraph\", \"s_icm.deduplicated.bedGraph\"]\n",
    "for file in files:\n",
    "    with open(file, \"r\") as f:\n",
    "        data = []\n",
    "        for line in f:\n",
    "            if line.split(\"\\t\")[0] == \"chr11\":\n",
    "                data.append(float(line.split(\"\\t\")[3][:-1]))\n",
    "        names, frequency = np.unique(data, return_counts=True)\n",
    "        plt.clf()\n",
    "        plt.xlabel(\"percentage\")\n",
    "        plt.ylabel(\"frequency\")\n",
    "        xn = [f\"{(i)*10}\" for i in range(11)]\n",
    "        xp = np.arange(0, len(frequency)+1, len(frequency)/10)\n",
    "        plt.xticks(xp, xn)\n",
    "        plt.title(file)\n",
    "        plt.plot(frequency/len(data))\n",
    "        plt.show()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
